<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Matthew A. Barbour" />

<meta name="date" content="2021-06-24" />

<title>Plant growth in the absence of insects</title>

<script src="site_libs/header-attrs-2.8/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">genes-to-foodweb-stability</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mabarbour/genes-to-foodweb-stability">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Plant growth in the absence of insects</h1>
<h4 class="author">Matthew A. Barbour</h4>
<h4 class="date">2021-06-24</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-06-24
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>genes-to-foodweb-stability/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200205code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200205)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200205code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200205)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommabarbourgenestofoodwebstabilitytree1955523a9c250979d19c40a7e59a9a41edfee38etargetblank1955523a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mabarbour/genes-to-foodweb-stability/tree/1955523a9c250979d19c40a7e59a9a41edfee38e" target="_blank">1955523</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommabarbourgenestofoodwebstabilitytree1955523a9c250979d19c40a7e59a9a41edfee38etargetblank1955523a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/mabarbour/genes-to-foodweb-stability/tree/1955523a9c250979d19c40a7e59a9a41edfee38e" target="_blank">1955523</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/.Rhistory
    Ignored:    output/.Rapp.history

Untracked files:
    Untracked:  .genes-to-foodweb-stability.Rproj.swp
    Untracked:  analysis/foodweb-persistence.Rmd
    Untracked:  analysis/foodweb-transitions.Rmd
    Untracked:  analysis/plant-growth-no-insects-lme.Rmd
    Untracked:  analysis/prep-timeseries.Rmd
    Untracked:  analysis/structural-stability.Rmd
    Untracked:  code/summarize_stability_metrics.R
    Untracked:  data/InsectAbundanceSurvival.csv
    Untracked:  data/insect_abundance_data_2018-09-26_underneathleaf.csv
    Untracked:  figures/MAR1-parameter-plot.pdf
    Untracked:  figures/aop2-genos-coxadjcurve.pdf
    Untracked:  figures/aop2-genotypes-multistate.pdf
    Untracked:  figures/cage-dynamics-20C.pdf
    Untracked:  figures/cage-dynamics-23C.pdf
    Untracked:  figures/keystone-coxadjcurve.pdf
    Untracked:  figures/keystone-effect-model-comparison-ratio.pdf
    Untracked:  output/all.mar1.brm.adj.rds
    Untracked:  output/all.mar1.brm.unadj.ar2.lag.rds
    Untracked:  output/all.mar1.brm.unadj.noBRBRonLYER.rds
    Untracked:  output/all.mar1.brm.unadj.rds
    Untracked:  output/all.mar1.brm.unadj.xAOP2.rds
    Untracked:  output/initial.mar1.brm.adj.rds
    Untracked:  output/initial.mar1.brm.unadj.rds
    Untracked:  output/time-series-data.RData

Unstaged changes:
    Modified:   README.md
    Deleted:    analysis/about.Rmd
    Deleted:    analysis/community-persistence-keystone.Rmd
    Deleted:    analysis/critical-transitions-keystone.Rmd
    Modified:   analysis/index.Rmd
    Deleted:    analysis/plant-growth-no-insects.Rmd
    Deleted:    analysis/structural-stability-keystone.Rmd
    Deleted:    code/AOP2-LYER-Ptoid-persistence.R
    Modified:   code/README.md
    Deleted:    code/glm-ftest.R
    Modified:   code/plot-feasibility-domain.R
    Deleted:    code/prep-time-series.R
    Modified:   code/simulate-community-dynamics.R
    Deleted:    code/temperature-structural-stability-fig.R
    Modified:   data/README.md
    Deleted:    data/arabidopsis_clean_df.csv
    Modified:   figures/AOP2-growth-no-insects.pdf
    Modified:   figures/MAR1-nonequilibrium-foodchain-AOP2.pdf
    Deleted:    figures/MAR1-nonequilibrium-initial-temp.pdf
    Modified:   figures/MAR1-posterior-foodchain-AOP2.pdf
    Deleted:    figures/MAR1-posterior-initial-temp.pdf
    Deleted:    figures/initial-foodweb-structural-stability.png
    Modified:   figures/keystone-gene.pdf
    Modified:   figures/keystone-structural-stability-forkeynote.pdf
    Modified:   output/README.md
    Deleted:    output/full.mv.norm.brm.keystone.rds
    Deleted:    output/plant-growth-no-insects.RData
    Deleted:    output/reduced.1.brm.keystone.rds
    Deleted:    output/reduced.2.brm.keystone.rds
    Deleted:    output/reduced.3.brm.keystone.rds
    Deleted:    output/reduced.4.brm.keystone.rds
    Deleted:    output/reduced.5.brm.keystone.rds
    Deleted:    output/reduced.6.brm.keystone.rds
    Deleted:    output/timeseries_df.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="wrangle-data" class="section level1">
<h1>Wrangle data</h1>
<pre class="r"><code># load data
ChamberNoInsectsDF &lt;- read_csv(&quot;data/PreExperimentNoInsectsPlantBiomass.csv&quot;) %&gt;%
  mutate(Cage = as.character(Cage),
         Pot = as.character(Pot))

# conduct analyses at cage level
CageLevelBiomass &lt;- ChamberNoInsectsDF %&gt;%
  # sum biomass across both pots
  group_by(Cage, Temperature, Richness, Composition, Col, gsm1, AOP2, AOP2.gsoh) %&gt;%
  summarise_at(vars(Biomass_g), list(sum)) %&gt;%
  # tidy data
  ungroup() %&gt;%
  select(cage = Cage, temp = Temperature, rich = Richness, com = Composition, Col, gsm1, AOP2, AOP2.gsoh, Biomass_g) %&gt;%
  # adjust temp and rich so effect of +1 C is comparable to +1 genotype
  mutate(temp = ifelse(temp == &quot;20 C&quot;, 0, 3),
         rich = rich - 1,
         # define orthogonal constrasts to test for above-average allele effects.
        # aop2_vs_AOP2 must be included first
        aop2_vs_AOP2 = Col + gsm1 - AOP2 - AOP2.gsoh,
        mam1_vs_MAM1 = gsm1 - Col, # aop2_vs_AOP2 must be included in model
        gsoh_vs_GSOH = AOP2.gsoh - AOP2)</code></pre>
</div>
<div id="reproduce-table-s5" class="section level1">
<h1>Reproduce Table S5</h1>
<pre class="r"><code># fit model
log_biomass_lmer &lt;- lmer(log(Biomass_g) ~ temp*(rich + aop2_vs_AOP2 + mam1_vs_MAM1 + gsoh_vs_GSOH) + (1|com/temp), 
                        data = CageLevelBiomass)

# reproduce table S5
anova(log_biomass_lmer, type = &quot;1&quot;, ddf = &quot;Kenward-Roger&quot;) %&gt;%
  kable(., caption = &quot;Analysis of variance for plant biomass (log transformed) in the absence of insects.&quot;, booktabs = T) %&gt;%
  kable_styling(latex_options = c(&quot;striped&quot;, &quot;hold_position&quot;))</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Analysis of variance for plant biomass (log transformed) in the absence of insects.
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Sum Sq
</th>
<th style="text-align:right;">
Mean Sq
</th>
<th style="text-align:right;">
NumDF
</th>
<th style="text-align:right;">
DenDF
</th>
<th style="text-align:right;">
F value
</th>
<th style="text-align:right;">
Pr(&gt;F)
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
temp
</td>
<td style="text-align:right;">
4.2342063
</td>
<td style="text-align:right;">
4.2342063
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
5.220228
</td>
<td style="text-align:right;">
52.8748142
</td>
<td style="text-align:right;">
0.0006393
</td>
</tr>
<tr>
<td style="text-align:left;">
rich
</td>
<td style="text-align:right;">
0.0110385
</td>
<td style="text-align:right;">
0.0110385
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
4.951544
</td>
<td style="text-align:right;">
0.1378439
</td>
<td style="text-align:right;">
0.7257894
</td>
</tr>
<tr>
<td style="text-align:left;">
aop2_vs_AOP2
</td>
<td style="text-align:right;">
1.2616246
</td>
<td style="text-align:right;">
1.2616246
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
5.791912
</td>
<td style="text-align:right;">
15.7545862
</td>
<td style="text-align:right;">
0.0079189
</td>
</tr>
<tr>
<td style="text-align:left;">
mam1_vs_MAM1
</td>
<td style="text-align:right;">
0.0410509
</td>
<td style="text-align:right;">
0.0410509
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
5.791912
</td>
<td style="text-align:right;">
0.5126251
</td>
<td style="text-align:right;">
0.5018356
</td>
</tr>
<tr>
<td style="text-align:left;">
gsoh_vs_GSOH
</td>
<td style="text-align:right;">
0.0148935
</td>
<td style="text-align:right;">
0.0148935
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
5.791912
</td>
<td style="text-align:right;">
0.1859833
</td>
<td style="text-align:right;">
0.6818684
</td>
</tr>
<tr>
<td style="text-align:left;">
temp:rich
</td>
<td style="text-align:right;">
0.0097817
</td>
<td style="text-align:right;">
0.0097817
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
4.427930
</td>
<td style="text-align:right;">
0.1221488
</td>
<td style="text-align:right;">
0.7427133
</td>
</tr>
<tr>
<td style="text-align:left;">
temp:aop2_vs_AOP2
</td>
<td style="text-align:right;">
0.0035946
</td>
<td style="text-align:right;">
0.0035946
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
5.647776
</td>
<td style="text-align:right;">
0.0448871
</td>
<td style="text-align:right;">
0.8396490
</td>
</tr>
<tr>
<td style="text-align:left;">
temp:mam1_vs_MAM1
</td>
<td style="text-align:right;">
0.0541709
</td>
<td style="text-align:right;">
0.0541709
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
5.647776
</td>
<td style="text-align:right;">
0.6764608
</td>
<td style="text-align:right;">
0.4441364
</td>
</tr>
<tr>
<td style="text-align:left;">
temp:gsoh_vs_GSOH
</td>
<td style="text-align:right;">
0.0669588
</td>
<td style="text-align:right;">
0.0669588
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
5.647776
</td>
<td style="text-align:right;">
0.8361506
</td>
<td style="text-align:right;">
0.3978575
</td>
</tr>
</tbody>
</table>
</div>
<div id="reproduce-fig.-s10" class="section level1">
<h1>Reproduce Fig. S10</h1>
<pre class="r"><code># calculate 95% confidence intervals, but first remove higher-order statistical interactions
summary(lme4::lmer(log(Biomass_g) ~ temp + rich + aop2_vs_AOP2 + (1|com), data = CageLevelBiomass)) # dropped com:temp, because it was singular, it also doesn&#39;t matter for estimating confidence intervals for genetic effect, it would matter if we were trying to estimate temp, which we are only controlling for here.</code></pre>
<pre><code>Linear mixed model fit by REML [&#39;lmerMod&#39;]
Formula: log(Biomass_g) ~ temp + rich + aop2_vs_AOP2 + (1 | com)
   Data: CageLevelBiomass

REML criterion at convergence: 39.1

Scaled residuals: 
    Min      1Q  Median      3Q     Max 
-2.7409 -0.4813  0.1360  0.5751  2.4735 

Random effects:
 Groups   Name        Variance Std.Dev.
 com      (Intercept) 0.009476 0.09735 
 Residual             0.081126 0.28483 
Number of obs: 60, groups:  com, 11

Fixed effects:
             Estimate Std. Error t value
(Intercept)  -0.05210    0.07843  -0.664
temp         -0.21085    0.02451  -8.601
rich         -0.02374    0.05456  -0.435
aop2_vs_AOP2  0.19691    0.04554   4.324

Correlation of Fixed Effects:
            (Intr) temp   rich  
temp        -0.469              
rich        -0.645  0.000       
ap2_vs_AOP2  0.000  0.000  0.000</code></pre>
<pre class="r"><code>aop2_CI &lt;- tidy(lmer(log(Biomass_g) ~ -1 + temp + I(AOP2 + AOP2.gsoh) + I(Col + gsm1) + (1|com), 
                     data = CageLevelBiomass), 
                conf.int = T, conf.level = 0.95) %&gt;%
  filter(term %in% c(&quot;I(AOP2 + AOP2.gsoh)&quot;,&quot;I(Col + gsm1)&quot;)) %&gt;%
  mutate(allele = c(&quot;AOP2&quot;,&quot;aop2&quot;))
# note that I back transform to original scale for plotting
exp(aop2_CI$estimate[2])</code></pre>
<pre><code>[1] 1.17452</code></pre>
<pre class="r"><code># get the effect of each genotype
mean_geno &lt;- tidy(lmer(log(Biomass_g) ~ -1 + temp + AOP2 + AOP2.gsoh + Col + gsm1 + (1|com), 
                     data = CageLevelBiomass), 
                conf.int = T, conf.level = 0.95) %&gt;%
  filter(term %in% c(&quot;AOP2&quot;,&quot;AOP2.gsoh&quot;,&quot;Col&quot;,&quot;gsm1&quot;)) %&gt;%
  mutate(allele = c(&quot;AOP2&quot;,&quot;AOP2&quot;,&quot;aop2&quot;,&quot;aop2&quot;),
         term = factor(term, levels = c(&quot;Col&quot;,&quot;gsm1&quot;,&quot;AOP2&quot;,&quot;AOP2.gsoh&quot;), labels = c(&quot;Col&quot;,&quot;gsm1&quot;,&quot;AOP2&quot;,&quot;AOP2/gsoh&quot;)))

# plot on original scale
# adding a genotype with an aop2 allele to the population doubles the likelihood of species persistence
plot_AOP2_growth_no_insects &lt;- ggplot(aop2_CI, aes(x = allele, y = exp(estimate))) +
  geom_point(size = 5) +
  geom_point(data = mean_geno, aes(color = term), size = 5, position = position_dodge(width = 0.3)) +
  geom_linerange(aes(ymax = exp(estimate + std.error), ymin = exp(estimate - std.error)), size = 1.5) +
  geom_linerange(aes(ymax = exp(conf.high), ymin = exp(conf.low))) +
  scale_x_discrete(labels = c(&quot;AOP2\u2013&quot;,&quot;AOP2+&quot;)) +
  scale_y_continuous(&quot;Plant biomass (g)&quot;) +
  xlab(&quot;Allele&quot;) + 
  scale_color_manual(values = c(&quot;darkgreen&quot;,&quot;steelblue&quot;,&quot;darkorange&quot;,&quot;firebrick1&quot;), name = &quot;&quot;) + 
  theme_cowplot(font_size = 18, line_size = 1)

x11(); plot_AOP2_growth_no_insects

ggsave(plot = plot_AOP2_growth_no_insects, filename = &quot;figures/AOP2-growth-no-insects.pdf&quot;, height = 6, width = 8, device=cairo_pdf)</code></pre>
<p><img src="figure/plant-growth-no-insects-lme.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.7 LTS

Matrix products: default
BLAS:   /usr/lib/libblas/libblas.so.3.6.0
LAPACK: /usr/lib/lapack/liblapack.so.3.6.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] broom.mixed_0.2.6 lmerTest_3.1-3    lme4_1.1-27       Matrix_1.3-4     
 [5] cowplot_1.1.1     kableExtra_1.3.4  forcats_0.5.1     stringr_1.4.0    
 [9] dplyr_1.0.6       purrr_0.3.4       readr_1.4.0       tidyr_1.1.3      
[13] tibble_3.1.2      ggplot2_3.3.3     tidyverse_1.3.1   workflowr_1.6.2  

loaded via a namespace (and not attached):
 [1] nlme_3.1-152        pbkrtest_0.5.1      fs_1.5.0           
 [4] lubridate_1.7.10    webshot_0.5.2       httr_1.4.2         
 [7] rprojroot_2.0.2     numDeriv_2016.8-1.1 tools_4.1.0        
[10] TMB_1.7.20          backports_1.2.1     bslib_0.2.5.1      
[13] utf8_1.2.1          R6_2.5.0            DBI_1.1.1          
[16] colorspace_2.0-1    withr_2.4.2         tidyselect_1.1.1   
[19] compiler_4.1.0      git2r_0.28.0        cli_2.5.0          
[22] rvest_1.0.0         xml2_1.3.2          labeling_0.4.2     
[25] sass_0.4.0          scales_1.1.1        systemfonts_1.0.2  
[28] digest_0.6.27       minqa_1.2.4         rmarkdown_2.8      
[31] svglite_1.2.3.2     pkgconfig_2.0.3     htmltools_0.5.1.1  
[34] highr_0.9           dbplyr_2.1.1        rlang_0.4.11       
[37] readxl_1.3.1        rstudioapi_0.13     farver_2.1.0       
[40] jquerylib_0.1.4     generics_0.1.0      jsonlite_1.7.2     
[43] magrittr_2.0.1      Rcpp_1.0.6          munsell_0.5.0      
[46] fansi_0.5.0         gdtools_0.2.3       lifecycle_1.0.0    
[49] stringi_1.6.2       yaml_2.2.1          MASS_7.3-54        
[52] plyr_1.8.6          grid_4.1.0          parallel_4.1.0     
[55] promises_1.2.0.1    crayon_1.4.1        lattice_0.20-44    
[58] haven_2.4.1         splines_4.1.0       hms_1.1.0          
[61] knitr_1.33          pillar_1.6.1        boot_1.3-28        
[64] reshape2_1.4.4      reprex_2.0.0        glue_1.4.2         
[67] evaluate_0.14       modelr_0.1.8        vctrs_0.3.8        
[70] nloptr_1.2.2.2      httpuv_1.6.1        cellranger_1.1.0   
[73] gtable_0.3.0        assertthat_0.2.1    xfun_0.23          
[76] broom_0.7.6         coda_0.19-4         later_1.2.0        
[79] viridisLite_0.4.0   ellipsis_0.3.2     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
