<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="mabarbour" />

<meta name="date" content="2021-05-19" />

<title>prep-timeseries</title>

<script src="site_libs/header-attrs-2.8/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">genes-to-foodweb-stability</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mabarbour/genes-to-foodweb-stability">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">prep-timeseries</h1>
<h4 class="author">mabarbour</h4>
<h4 class="date">2021-05-19</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-06-24
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>genes-to-foodweb-stability/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200205code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200205)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200205code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200205)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommabarbourgenestofoodwebstabilitytree1955523a9c250979d19c40a7e59a9a41edfee38etargetblank1955523a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mabarbour/genes-to-foodweb-stability/tree/1955523a9c250979d19c40a7e59a9a41edfee38e" target="_blank">1955523</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommabarbourgenestofoodwebstabilitytree1955523a9c250979d19c40a7e59a9a41edfee38etargetblank1955523a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/mabarbour/genes-to-foodweb-stability/tree/1955523a9c250979d19c40a7e59a9a41edfee38e" target="_blank">1955523</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/.Rhistory
    Ignored:    output/.Rapp.history

Untracked files:
    Untracked:  .genes-to-foodweb-stability.Rproj.swp
    Untracked:  analysis/foodweb-persistence.Rmd
    Untracked:  analysis/foodweb-transitions.Rmd
    Untracked:  analysis/plant-growth-no-insects-lme.Rmd
    Untracked:  analysis/prep-timeseries.Rmd
    Untracked:  analysis/structural-stability.Rmd
    Untracked:  code/summarize_stability_metrics.R
    Untracked:  data/InsectAbundanceSurvival.csv
    Untracked:  data/insect_abundance_data_2018-09-26_underneathleaf.csv
    Untracked:  figures/MAR1-parameter-plot.pdf
    Untracked:  figures/aop2-genos-coxadjcurve.pdf
    Untracked:  figures/aop2-genotypes-multistate.pdf
    Untracked:  figures/cage-dynamics-20C.pdf
    Untracked:  figures/cage-dynamics-23C.pdf
    Untracked:  figures/keystone-coxadjcurve.pdf
    Untracked:  figures/keystone-effect-model-comparison-ratio.pdf
    Untracked:  output/all.mar1.brm.adj.rds
    Untracked:  output/all.mar1.brm.unadj.ar2.lag.rds
    Untracked:  output/all.mar1.brm.unadj.noBRBRonLYER.rds
    Untracked:  output/all.mar1.brm.unadj.rds
    Untracked:  output/all.mar1.brm.unadj.xAOP2.rds
    Untracked:  output/initial.mar1.brm.adj.rds
    Untracked:  output/initial.mar1.brm.unadj.rds
    Untracked:  output/time-series-data.RData

Unstaged changes:
    Modified:   README.md
    Deleted:    analysis/about.Rmd
    Deleted:    analysis/community-persistence-keystone.Rmd
    Deleted:    analysis/critical-transitions-keystone.Rmd
    Modified:   analysis/index.Rmd
    Deleted:    analysis/plant-growth-no-insects.Rmd
    Deleted:    analysis/structural-stability-keystone.Rmd
    Deleted:    code/AOP2-LYER-Ptoid-persistence.R
    Modified:   code/README.md
    Deleted:    code/glm-ftest.R
    Modified:   code/plot-feasibility-domain.R
    Deleted:    code/prep-time-series.R
    Modified:   code/simulate-community-dynamics.R
    Deleted:    code/temperature-structural-stability-fig.R
    Modified:   data/README.md
    Deleted:    data/arabidopsis_clean_df.csv
    Modified:   figures/AOP2-growth-no-insects.pdf
    Modified:   figures/MAR1-nonequilibrium-foodchain-AOP2.pdf
    Deleted:    figures/MAR1-nonequilibrium-initial-temp.pdf
    Modified:   figures/MAR1-posterior-foodchain-AOP2.pdf
    Deleted:    figures/MAR1-posterior-initial-temp.pdf
    Deleted:    figures/initial-foodweb-structural-stability.png
    Modified:   figures/keystone-gene.pdf
    Modified:   figures/keystone-structural-stability-forkeynote.pdf
    Modified:   output/README.md
    Deleted:    output/full.mv.norm.brm.keystone.rds
    Deleted:    output/plant-growth-no-insects.RData
    Deleted:    output/reduced.1.brm.keystone.rds
    Deleted:    output/reduced.2.brm.keystone.rds
    Deleted:    output/reduced.3.brm.keystone.rds
    Deleted:    output/reduced.4.brm.keystone.rds
    Deleted:    output/reduced.5.brm.keystone.rds
    Deleted:    output/reduced.6.brm.keystone.rds
    Deleted:    output/timeseries_df.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="upload-original-data" class="section level1">
<h1>Upload original data</h1>
<pre class="r"><code>original_df &lt;- read_csv(&quot;data/InsectAbundanceSurvival.csv&quot;)</code></pre>
</div>
<div id="underneath-leaves-correction-for-lyer-and-ptoid" class="section level1">
<h1>Underneath leaves correction for LYER and Ptoid</h1>
<p>We are only using data from the last week of the experiment, for which we have estimates of LYER and Ptoids (mummies actually) underneath the leaf</p>
<pre class="r"><code># only retain cages where LYER or Parasitoids survived
# no surviving BRBR, so we can&#39;t estimate this
LYER_df_17 &lt;- filter(original_df, Week == &quot;17&quot;) %&gt;% filter(LYER_Survival == 1) %&gt;% dplyr::select(Cage, LYER)
Mummy_df_17 &lt;- filter(original_df, Week == &quot;17&quot;) %&gt;% filter(Mummy_Ptoids_Survival == 1) %&gt;% dplyr::select(Cage, Mummy_raw)

# underneath leaf counts, only available for last week of experiment
underneath_df &lt;- read_csv(&quot;data/insect_abundance_data_2018-09-26_underneathleaf.csv&quot;) # counted to individual resolution

# merge with original data
LYER_df_17 &lt;- left_join(LYER_df_17, dplyr::select(underneath_df, Cage, underneath_LYER = LYER))
Mummy_df_17 &lt;- left_join(Mummy_df_17, dplyr::select(underneath_df, Cage, underneath_Mummy = Mummy))</code></pre>
<div id="lyer" class="section level2">
<h2>LYER</h2>
<pre class="r"><code>## exploratory plots ----

# all data
x11(); ggplot(LYER_df_17, aes(x = log1p(LYER), y = log(underneath_LYER))) +
  geom_point() + geom_smooth(method = &quot;lm&quot;) 

# remove outlier
x11(); ggplot(LYER_df_17[-31,], aes(x = log1p(LYER), y = log(underneath_LYER))) +
  geom_point() + geom_smooth(method = &quot;lm&quot;)

## GLM ----

# I dropped the outlier because I truly think it was an unusual data point and would bias our other estimates.
plot(MASS::glm.nb(underneath_LYER ~ log1p(LYER), LYER_df_17), which = 5)</code></pre>
<p><img src="figure/prep-timeseries.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/prep-timeseries.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>summary(glm(underneath_LYER ~ log1p(LYER), LYER_df_17[-31,], family = poisson)) # residual deviance suggests a lot of overdispersion, so I&#39;m going to use a negative binomial model</code></pre>
<pre><code>
Call:
glm(formula = underneath_LYER ~ log1p(LYER), family = poisson, 
    data = LYER_df_17[-31, ])

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-6.9703  -3.1734  -0.9052   2.8304   7.9681  

Coefficients:
            Estimate Std. Error z value Pr(&gt;|z|)    
(Intercept)  3.47991    0.05716   60.88   &lt;2e-16 ***
log1p(LYER)  0.29613    0.01141   25.95   &lt;2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

(Dispersion parameter for poisson family taken to be 1)

    Null deviance: 1316.37  on 30  degrees of freedom
Residual deviance:  481.88  on 29  degrees of freedom
AIC: 683.74

Number of Fisher Scoring iterations: 5</code></pre>
<pre class="r"><code>LYER_underneath_glm.nb &lt;- MASS::glm.nb(underneath_LYER ~ log1p(LYER), LYER_df_17[-31,]) 
summary(LYER_underneath_glm.nb)</code></pre>
<pre><code>
Call:
MASS::glm.nb(formula = underneath_LYER ~ log1p(LYER), data = LYER_df_17[-31, 
    ], init.theta = 5.709483328, link = log)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-2.2762  -0.7531  -0.2120   0.6421   1.9575  

Coefficients:
            Estimate Std. Error z value Pr(&gt;|z|)    
(Intercept)  3.41254    0.18541  18.405  &lt; 2e-16 ***
log1p(LYER)  0.31152    0.04166   7.477 7.59e-14 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

(Dispersion parameter for Negative Binomial(5.7095) family taken to be 1)

    Null deviance: 80.035  on 30  degrees of freedom
Residual deviance: 32.974  on 29  degrees of freedom
AIC: 327.12

Number of Fisher Scoring iterations: 1

              Theta:  5.71 
          Std. Err.:  1.57 

 2 x log-likelihood:  -321.123 </code></pre>
<pre class="r"><code>LYER_correction_glm.nb &lt;- function(x) round(predict(LYER_underneath_glm.nb, newdata = data.frame(LYER = x), type = &quot;response&quot;), 0)
# will add between 30 and 286 individual LYER to LYER_t
LYER_correction_glm.nb(c(0,max(original_df$LYER, na.rm = T)))</code></pre>
<pre><code>  1   2 
 30 286 </code></pre>
</div>
<div id="mummy-and-ptoids" class="section level2">
<h2>Mummy and Ptoids</h2>
<pre class="r"><code>## exploratory plot ----
x11(); ggplot(Mummy_df_17, aes(x = log1p(Mummy_raw), y = log(underneath_Mummy))) +
  geom_point() + geom_smooth(method = &quot;lm&quot;)
# the predicted increase is based on 1 data point
summary(Mummy_outlier_glm.nb &lt;- MASS::glm.nb(underneath_Mummy ~ log1p(Mummy_raw), Mummy_df_17))</code></pre>
<p><img src="figure/prep-timeseries.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
Call:
MASS::glm.nb(formula = underneath_Mummy ~ log1p(Mummy_raw), data = Mummy_df_17, 
    init.theta = 2.736861092, link = log)

Deviance Residuals: 
       1         2         3         4         5         6         7  
 0.00000  -0.95506  -1.45780   1.13467   0.04516   1.13467  -1.45780  

Coefficients:
                 Estimate Std. Error z value Pr(&gt;|z|)    
(Intercept)        1.5755     0.3088   5.102 3.37e-07 ***
log1p(Mummy_raw)   1.5279     0.2968   5.149 2.62e-07 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

(Dispersion parameter for Negative Binomial(2.7369) family taken to be 1)

    Null deviance: 48.8886  on 6  degrees of freedom
Residual deviance:  7.7395  on 5  degrees of freedom
AIC: 48.722

Number of Fisher Scoring iterations: 1

              Theta:  2.74 
          Std. Err.:  2.45 

 2 x log-likelihood:  -42.722 </code></pre>
<pre class="r"><code># hmmm, this model predicts that there could be as many as 43,291 mummies underneath the leaves, which is insane
round(predict(Mummy_outlier_glm.nb, newdata = data.frame(Mummy_raw = c(0, max(original_df$Mummy_raw, na.rm = T))), type = &quot;response&quot;),0)</code></pre>
<pre><code>    1     2 
    5 43291 </code></pre>
<pre class="r"><code># I think the baseline value, for which we have 5 data points is reasonable
# but for the increase, I&#39;m going to use the slope of LYER, which seems reasonable given that this was the primary
# consumer-resource relationship and LYER was often found underneath the leaves compared to BRBR

summary(Mummy_underneath_glm.nb &lt;- MASS::glm.nb(underneath_Mummy ~ 1, filter(Mummy_df_17, Mummy_raw == 0)))</code></pre>
<pre><code>
Call:
MASS::glm.nb(formula = underneath_Mummy ~ 1, data = filter(Mummy_df_17, 
    Mummy_raw == 0), init.theta = 1.874400108, link = log)

Deviance Residuals: 
       1         2         3         4         5         6  
-0.85061  -1.31170   0.98867   0.03969   0.98867  -1.31170  

Coefficients:
            Estimate Std. Error z value Pr(&gt;|z|)    
(Intercept)   1.5755     0.3513   4.485 7.29e-06 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

(Dispersion parameter for Negative Binomial(1.8744) family taken to be 1)

    Null deviance: 6.1212  on 5  degrees of freedom
Residual deviance: 6.1212  on 5  degrees of freedom
AIC: 35.421

Number of Fisher Scoring iterations: 1

              Theta:  1.87 
          Std. Err.:  1.51 

 2 x log-likelihood:  -31.421 </code></pre>
<pre class="r"><code>round(exp(coef(Mummy_underneath_glm.nb)[&quot;(Intercept)&quot;]),0) # predicts 5 underneath leaves if there was a zero above rosette count</code></pre>
<pre><code>(Intercept) 
          5 </code></pre>
<pre class="r"><code>Mummy_correction_glm.nb &lt;- function(x) round(exp(coef(Mummy_underneath_glm.nb)[&quot;(Intercept)&quot;] + coef(LYER_underneath_glm.nb)[&quot;log1p(LYER)&quot;]*log1p(x)),0)
Mummy_correction_glm.nb(c(0,max(original_df$Mummy_raw, na.rm = T))) # now that&#39;s more reasonable</code></pre>
<pre><code>[1]  5 31</code></pre>
</div>
</div>
<div id="measurement-error" class="section level1">
<h1>Measurement error</h1>
<p>Below, I estimate measurement error for BRBR, LYER, and Ptoid counts. This uses data for which we had overlapping counts for MAB (first author) and DTV (technician that was instrumental to the study).</p>
<pre class="r"><code># identify cage-week combinations with repeated measurements
rep_df &lt;- original_df %&gt;% 
  distinct(Cage, Week, Counter) %&gt;%
  group_by(Cage, Week) %&gt;%
  summarise(n = n()) %&gt;%
  filter(n &gt; 1)</code></pre>
<div id="brbr" class="section level2">
<h2>BRBR</h2>
<pre class="r"><code># make BRBR measurement error (me) df
BRBR_me_df &lt;- left_join(rep_df, dplyr::select(original_df, Cage, Week, Counter, BRBR, BRBR_Survival)) %&gt;% 
  drop_na() %&gt;% 
  filter(BRBR_Survival == 1) %&gt;%
  spread(key = Counter, value = BRBR) %&gt;%
  mutate(log1p_DTV = log1p(DTV),
         log1p_MAB = log1p(MAB)) %&gt;%
  ungroup()
BRBR_me &lt;- apply(select(BRBR_me_df, log1p_DTV, log1p_MAB), MARGIN = 1, FUN = sd)
hist(BRBR_me)</code></pre>
<p><img src="figure/prep-timeseries.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>median(BRBR_me, na.rm = T)</code></pre>
<pre><code>[1] 0.2829706</code></pre>
</div>
<div id="lyer-1" class="section level2">
<h2>LYER</h2>
<pre class="r"><code># make LYER measurement error (me) df
LYER_me_df &lt;- left_join(rep_df, dplyr::select(original_df, Cage, Week, Counter, LYER, LYER_Survival)) %&gt;% 
  drop_na() %&gt;% 
  filter(LYER_Survival == 1) %&gt;%
  spread(key = Counter, value = LYER) %&gt;%
  mutate(log1p_DTV = log1p(DTV),
         log1p_MAB = log1p(MAB),
         abs.diff_log1p = log1p_DTV - log1p_MAB,
         mean.diff_log1p = log1p_DTV - (log1p_DTV+log1p_MAB)/2) %&gt;%
  ungroup()
LYER_me &lt;- apply(select(LYER_me_df, log1p_DTV, log1p_MAB), MARGIN = 1, FUN = sd)
hist(LYER_me)</code></pre>
<p><img src="figure/prep-timeseries.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>median(LYER_me, na.rm = T)</code></pre>
<pre><code>[1] 0.2433557</code></pre>
</div>
<div id="mummy-ptoids" class="section level2">
<h2>Mummy-Ptoids</h2>
<pre class="r"><code># make Mummy_Ptoids measurement error (me) df
Mummy_Ptoids_me_df &lt;- left_join(rep_df, dplyr::select(original_df, Cage, Week, Counter, Mummy_Ptoids, Mummy_Ptoids_Survival)) %&gt;% 
  drop_na() %&gt;% 
  filter(Mummy_Ptoids_Survival == 1) %&gt;%
  spread(key = Counter, value = Mummy_Ptoids) %&gt;%
  mutate(log1p_DTV = log1p(DTV),
         log1p_MAB = log1p(MAB)) %&gt;%
  ungroup()
Mummy_Ptoids_me &lt;- apply(select(Mummy_Ptoids_me_df, log1p_DTV, log1p_MAB), MARGIN = 1, FUN = sd)
hist(Mummy_Ptoids_me)</code></pre>
<p><img src="figure/prep-timeseries.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>median(Mummy_Ptoids_me, na.rm = T)</code></pre>
<pre><code>[1] 0.1323786</code></pre>
</div>
</div>
<div id="tidy-timeseries-data" class="section level1">
<h1>Tidy timeseries data</h1>
<pre class="r"><code># useful function
mean_integer &lt;- function(x) as.integer(mean(x, na.rm=T)) # remove NA, otherwise, the count will be set to zero if only 1 counter counted.</code></pre>
<pre class="r"><code># make timeseries dataset
timeseries_df &lt;- original_df %&gt;%
  select(Cage, Week, temp = Temperature, rich = Richness, com = Composition, Col, gsm1, AOP2, AOP2.gsoh, BRBR_Survival, LYER_Survival, Mummy_Ptoids_Survival, BRBR_t = BRBR, LYER_t = LYER, Ptoid_t = Mummy_Ptoids, Mummy_t_raw = Mummy_raw) %&gt;%
  # applying adjustments before averaging across counters
  mutate(LYER_t_adj = ifelse(LYER_Survival == 0 | is.na(LYER_Survival) == T, 1,   # replace extinction zeros with value of 1
                             ifelse(LYER_Survival == 1, LYER_t + LYER_correction_glm.nb(LYER_t), LYER_t)), # add adjusted values
         # Ptoid should always be zero for Week 1 and 2, as they were added after sampling aphids in week 2. But we set them to 1 for extrapolating predictions
         Ptoid_t_adj = ifelse(Mummy_Ptoids_Survival == 0 | is.na(Mummy_Ptoids_Survival) == T, 1,   # replace extinction zeros with value of 1
                              ifelse(Mummy_Ptoids_Survival == 1, Ptoid_t + Mummy_correction_glm.nb(Mummy_t_raw), Ptoid_t))) %&gt;% # add adjusted values
  # apply adjustments for log-log analysis, using minimum count resolutions
  # note that we NEVER used extinct species in fitting the multivariate model, only for extrapolating predictions from multivariate model to extant species
  mutate(BRBR_t = ifelse(BRBR_Survival == 0 | is.na(BRBR_Survival) == T, 1,  # replace extinction zeros with value of 1
                         ifelse(BRBR_t == 0 &amp; BRBR_Survival == 1 &amp; Week &lt; 2, 1,   # replaced undetected zeros before week 2 with value of 1
                                ifelse(BRBR_t == 0 &amp; BRBR_Survival == 1 &amp; Week &gt; 1, 5, BRBR_t))), # replace undetected zeros with value of 5
         LYER_t = ifelse(LYER_Survival == 0 | is.na(LYER_Survival) == T, 1,   # replace extinction zeros with value of 1
                         ifelse(LYER_t == 0 &amp; LYER_Survival == 1 &amp; Week &lt; 2, 1,     # replaced undetected zeros before week 2 with value of 1
                                ifelse(LYER_t == 0 &amp; LYER_Survival == 1 &amp; Week &gt; 1, 5, LYER_t))), # replaced undetected zeros after week 1 with value of 5
         Ptoid_t = ifelse(Mummy_Ptoids_Survival == 0 | is.na(Mummy_Ptoids_Survival) == T, 1,   # replace extinction zeros with value of 1
                          ifelse(Ptoid_t == 0 &amp; Mummy_Ptoids_Survival == 1, 1, Ptoid_t))) %&gt;%     # replaced undetected zeros with value of 1
  group_by(Cage, Week, temp, rich, com, Col, gsm1, AOP2, AOP2.gsoh) %&gt;%
  # average across counters
  summarise_all(~ mean_integer(.)) %&gt;% # funs(mean_integer)
  ungroup()

# initial additions of aphids (4 individuals)
init_timeseries_df &lt;- timeseries_df %&gt;%
  distinct(Cage, temp, rich, com, Col, gsm1, AOP2, AOP2.gsoh) %&gt;%
  # set Ptoid values to 1 for extrapolating predictions
  mutate(Week = 0, BRBR_Survival = 1, LYER_Survival = 1, Mummy_Ptoids_Survival = NA, 
         BRBR_t = 4, LYER_t = 4, Ptoid_t = 1, Mummy_t_raw = NA, # set Mummy_t to be consistent with its raw classification in the data
         LYER_t_adj = 4, Ptoid_t_adj = 1)
init_timeseries_df </code></pre>
<pre><code># A tibble: 60 x 18
    Cage temp   rich com           Col  gsm1  AOP2 AOP2.gsoh  Week BRBR_Survival
   &lt;dbl&gt; &lt;chr&gt; &lt;dbl&gt; &lt;chr&gt;       &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt;         &lt;dbl&gt;
 1     1 20 C      2 AOP2_AOP2.…     0     0     1         1     0             1
 2     2 20 C      4 Poly            1     1     1         1     0             1
 3     3 20 C      2 gsm1_AOP2.…     0     1     0         1     0             1
 4     4 20 C      1 AOP2            0     0     1         0     0             1
 5     5 20 C      1 gsm1            0     1     0         0     0             1
 6     6 20 C      2 Col_AOP2        1     0     1         0     0             1
 7     7 20 C      4 Poly            1     1     1         1     0             1
 8     8 20 C      2 Col_AOP2.g…     1     0     0         1     0             1
 9     9 20 C      2 Col_gsm1        1     1     0         0     0             1
10    10 20 C      2 gsm1_AOP2.…     0     1     0         1     0             1
# … with 50 more rows, and 8 more variables: LYER_Survival &lt;dbl&gt;,
#   Mummy_Ptoids_Survival &lt;lgl&gt;, BRBR_t &lt;dbl&gt;, LYER_t &lt;dbl&gt;, Ptoid_t &lt;dbl&gt;,
#   Mummy_t_raw &lt;lgl&gt;, LYER_t_adj &lt;dbl&gt;, Ptoid_t_adj &lt;dbl&gt;</code></pre>
<pre class="r"><code># create lagged (t-1) and leading variable at time t+1
timeseries_df &lt;- bind_rows(init_timeseries_df, timeseries_df) %&gt;%
  arrange(Cage, Week) %&gt;%
  group_by(Cage) %&gt;%
  # leading
  mutate(BRBR_Survival_t1 = lead(BRBR_Survival, n = 1, order_by = Week),
         BRBR_t1 = lead(BRBR_t, n = 1, order_by = Week),
         LYER_Survival_t1 = lead(LYER_Survival, n = 1, order_by = Week),
         LYER_t1 = lead(LYER_t, n = 1, order_by = Week),
         LYER_t1_adj = lead(LYER_t_adj, n = 1, order_by = Week),
         Mummy_Ptoids_Survival_t1 = lead(Mummy_Ptoids_Survival, n = 1, order_by = Week),
         Ptoid_t1 = lead(Ptoid_t, n = 1, order_by = Week),
         Ptoid_t1_adj = lead(Ptoid_t_adj, n = 1, order_by = Week)) %&gt;%
  # lagged. set default = 1 to include in extrapolations for log analysis
  mutate(BRBR_t0 = lag(BRBR_t, n = 1, order_by = Week, default = 1),
         LYER_t0 = lag(LYER_t, n = 1, order_by = Week, default = 1),
         #LYER_t0_adj = lag(LYER_t_adj, n = 1, order_by = Week),
         Ptoid_t0 = lag(Ptoid_t, n = 1, order_by = Week, default = 1))#,
         #Ptoid_t0_adj = lag(Ptoid_t_adj, n = 1, order_by = Week))

# add two parasitoid females to experiment after counts on week 2 and 3
timeseries_df$Ptoid_t[which(timeseries_df$Week == 2)] &lt;- 2
timeseries_df$Ptoid_t[which(timeseries_df$Week == 3)] &lt;- timeseries_df$Ptoid_t[which(timeseries_df$Week == 3)] + 2
# repeat for adjusted
timeseries_df$Ptoid_t_adj[which(timeseries_df$Week == 2)] &lt;- 2
timeseries_df$Ptoid_t_adj[which(timeseries_df$Week == 3)] &lt;- timeseries_df$Ptoid_t_adj[which(timeseries_df$Week == 3)] + 2
# and for lagged values
timeseries_df$Ptoid_t0[which(timeseries_df$Week == 2)] &lt;- 1 # actual zero, but setting to 1 for log() analysis later.
timeseries_df$Ptoid_t0[which(timeseries_df$Week == 3)] &lt;- 2 # 2 parasitoids added in previous week
timeseries_df$Ptoid_t0[which(timeseries_df$Week == 4)] &lt;- timeseries_df$Ptoid_t0[which(timeseries_df$Week == 4)] + 2 # adding 2 parasitoids to count from previous week (often zero)
# and for lagged adjusted values
#timeseries_df$Ptoid_t0_adj[which(timeseries_df$Week == 2)] &lt;- 0 # maintaing at zero, because it is, will adjust by adding 1 for log() analysis later
#timeseries_df$Ptoid_t0_adj[which(timeseries_df$Week == 3)] &lt;- 2 # 2 parasitoids added in previous week
#timeseries_df$Ptoid_t0_adj[which(timeseries_df$Week == 4)] &lt;- timeseries_df$Ptoid_t0_adj[which(timeseries_df$Week == 4)] + 2 # adding 2 parasitoids to count from previous week (often zero)
# before we were ignoring the initial increase of parasitoids from week 2 to week 3, since they are at zeros in the dataset, although I added 2 individuals after week 2 and 3 counts. I added the lines above to incorporate this info
# note that Ptoid_t1 will not exactly match Ptoid_t, but that is intentional since those data are derived from counts

# adjust Mummy_Ptoid_Survival so it doesn&#39;t start at Week 3, but Week 2
timeseries_df$Mummy_Ptoids_Survival[which(timeseries_df$Week == 2)] &lt;- 1

# all NA for prediction next week at week 17, which makes sense because that&#39;s when the experiment ended so we are going to remove this week
filter(timeseries_df, Week == 17) %&gt;%
  select(BRBR_t1, LYER_t1, LYER_t1_adj, Ptoid_t1, Ptoid_t1_adj) %&gt;%
  data.frame()</code></pre>
<pre><code>   Cage BRBR_t1 LYER_t1 LYER_t1_adj Ptoid_t1 Ptoid_t1_adj
1     1      NA      NA          NA       NA           NA
2     2      NA      NA          NA       NA           NA
3     3      NA      NA          NA       NA           NA
4     4      NA      NA          NA       NA           NA
5     5      NA      NA          NA       NA           NA
6     6      NA      NA          NA       NA           NA
7     7      NA      NA          NA       NA           NA
8     8      NA      NA          NA       NA           NA
9     9      NA      NA          NA       NA           NA
10   10      NA      NA          NA       NA           NA
11   11      NA      NA          NA       NA           NA
12   12      NA      NA          NA       NA           NA
13   13      NA      NA          NA       NA           NA
14   14      NA      NA          NA       NA           NA
15   15      NA      NA          NA       NA           NA
16   16      NA      NA          NA       NA           NA
17   17      NA      NA          NA       NA           NA
18   18      NA      NA          NA       NA           NA
19   19      NA      NA          NA       NA           NA
20   20      NA      NA          NA       NA           NA
21   21      NA      NA          NA       NA           NA
22   22      NA      NA          NA       NA           NA
23   23      NA      NA          NA       NA           NA
24   24      NA      NA          NA       NA           NA
25   25      NA      NA          NA       NA           NA
26   26      NA      NA          NA       NA           NA
27   27      NA      NA          NA       NA           NA
28   28      NA      NA          NA       NA           NA
29   29      NA      NA          NA       NA           NA
30   30      NA      NA          NA       NA           NA
31   31      NA      NA          NA       NA           NA
32   32      NA      NA          NA       NA           NA
33   33      NA      NA          NA       NA           NA
34   34      NA      NA          NA       NA           NA
35   35      NA      NA          NA       NA           NA
36   36      NA      NA          NA       NA           NA
37   37      NA      NA          NA       NA           NA
38   38      NA      NA          NA       NA           NA
39   39      NA      NA          NA       NA           NA
40   40      NA      NA          NA       NA           NA
41   41      NA      NA          NA       NA           NA
42   42      NA      NA          NA       NA           NA
43   43      NA      NA          NA       NA           NA
44   44      NA      NA          NA       NA           NA
45   45      NA      NA          NA       NA           NA
46   46      NA      NA          NA       NA           NA
47   47      NA      NA          NA       NA           NA
48   48      NA      NA          NA       NA           NA
49   49      NA      NA          NA       NA           NA
50   50      NA      NA          NA       NA           NA
51   51      NA      NA          NA       NA           NA
52   52      NA      NA          NA       NA           NA
53   53      NA      NA          NA       NA           NA
54   54      NA      NA          NA       NA           NA
55   55      NA      NA          NA       NA           NA
56   56      NA      NA          NA       NA           NA
57   57      NA      NA          NA       NA           NA
58   58      NA      NA          NA       NA           NA
59   59      NA      NA          NA       NA           NA
60   60      NA      NA          NA       NA           NA</code></pre>
<pre class="r"><code>timeseries_df &lt;- filter(timeseries_df, Week != 17)

# 17 weeks of data still because we include initial Week = 0
dim(timeseries_df)[1] / length(unique(timeseries_df$Cage)) </code></pre>
<pre><code>[1] 17</code></pre>
<pre class="r"><code># add new predictors
timeseries_df &lt;- timeseries_df %&gt;%
  mutate(temp = ifelse(temp == &quot;20 C&quot;, 0, 3), # now on scale of 1 deg C
         aop2_vs_AOP2 = Col + gsm1 - AOP2 - AOP2.gsoh,
         aop2_genotypes = Col + gsm1,
         AOP2_genotypes = AOP2 + AOP2.gsoh) </code></pre>
</div>
<div id="plant-biomass-data" class="section level1">
<h1>Plant biomass data</h1>
<pre class="r"><code># load and manage plant biomass time-series data
tmp.biomass &lt;- read_csv(&quot;data/ExperimentPlantBiomass.csv&quot;) %&gt;%
  select(Cage, Week, No_Plants, Biomass_g) # No_Plants not necessary because we&#39;re using Biomass_g as a covariate rather than a response
  # note that there is no biomass data on first week, because plants were left in for 2 weeks
  # we are going to interpolate this biomass based on the plants growth at
  # week 2 of the experiment

# get biomass at t+1
get_biomass_t1 &lt;- tmp.biomass %&gt;%
  mutate(Week = Week - 1) %&gt;%
  select(Cage, Week, Biomass_g_t1 = Biomass_g)

# merge biomass at time t and t+1
merge.biomass.df &lt;- left_join(get_biomass_t1, tmp.biomass) %&gt;%
  # assuming a linear relationship in plant growth between weeks, since
  # they had not hit the saturation point on their growth curve
  mutate(Biomass_g = ifelse(Week == 1, Biomass_g_t1 - Biomass_g_t1/6, Biomass_g),
         No_Plants = ifelse(Week == 1, 8, No_Plants)) %&gt;%
  arrange(Cage, Week)

# infer biomass data at week 0
biomass.0.df &lt;- merge.biomass.df %&gt;%
  mutate(Week = Week - 1) %&gt;%
  select(Cage, Week, Biomass_g_t1 = Biomass_g) %&gt;%
  filter(Week == 0) %&gt;%
  mutate(Biomass_g = Biomass_g_t1 - Biomass_g_t1/5,
         No_Plants = 8)

# merge week 0 and rest of biomass data
biomass.df &lt;- bind_rows(filter(merge.biomass.df, Week &gt; 0), biomass.0.df) %&gt;%
  arrange(Cage, Week)</code></pre>
<div id="final-adjustments" class="section level2">
<h2>Final adjustments</h2>
<pre class="r"><code>timeseries_df &lt;- left_join(timeseries_df, biomass.df) %&gt;% # add plant biomass to timeseries data
  # turned logged predictors into named variables for measurement error model
  mutate(logBRBR_t0 = log(BRBR_t0),
         logBRBR_t = log(BRBR_t),
         logLYER_t0 = log(LYER_t0),
         logLYER_t = log(LYER_t),
         logLYER_t_adj = log(LYER_t_adj),
         logPtoid_t0 = log(Ptoid_t0),
         logPtoid_t = log(Ptoid_t),
         logPtoid_t_adj = log(Ptoid_t_adj)) %&gt;%
  # add measurement error. if alive, used median observed measurement error, if not, set to small value (0.01) to allow measurement error model to extrapolate
  mutate(se_logBRBRt = ifelse(BRBR_Survival == 0 | is.na(BRBR_Survival) == T, 0.01,
                              ifelse(BRBR_Survival == 1, median(BRBR_me, na.rm = T), NA)),
         se_logLYERt = ifelse(LYER_Survival == 0 | is.na(LYER_Survival) == T, 0.01,
                              ifelse(LYER_Survival == 1, median(LYER_me, na.rm = T), NA)),
         se_logPtoidt = ifelse(Mummy_Ptoids_Survival == 0 | is.na(Mummy_Ptoids_Survival) == T, 0.01,
                               ifelse(Mummy_Ptoids_Survival == 1, median(Mummy_Ptoids_me, na.rm = T), NA)))</code></pre>
</div>
</div>
<div id="subset-for-multivariate-analysis" class="section level1">
<h1>Subset for multivariate analysis</h1>
<div id="aphids-only" class="section level2">
<h2>Aphids only</h2>
<pre class="r"><code># filter first two weeks of experiment
aphids_only_df &lt;- filter(timeseries_df, Week %in% c(0,1))</code></pre>
</div>
<div id="all-species" class="section level2">
<h2>All species</h2>
<p>I only retain data for which all species had positive abundances at the previous time step, and after all species have been added.</p>
<pre class="r"><code># filter data where all species survived to time t
full_df &lt;- filter(timeseries_df, BRBR_Survival == 1, LYER_Survival == 1, Mummy_Ptoids_Survival == 1) 

# tabulate variation in timeseries length for each cage
with(full_df, table(Week, Cage))</code></pre>
<pre><code>    Cage
Week 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28
   2 1 1 1 1 1 1 1 1 1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1
   3 1 1 1 1 1 1 1 1 1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1
   4 1 1 1 1 1 1 1 1 1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1
   5 1 1 1 1 1 1 1 1 1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1
   6 1 1 1 1 1 1 1 0 1  1  1  1  0  1  1  1  1  1  1  1  1  1  1  1  0  1  1  0
   7 0 1 0 0 1 0 0 0 0  0  0  0  0  1  0  0  1  0  0  0  0  0  0  1  0  0  0  0
   8 0 1 0 0 0 0 0 0 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
    Cage
Week 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53
   2  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1
   3  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1
   4  1  1  1  1  1  1  1  0  1  0  1  0  1  1  1  1  1  1  0  1  1  1  1  1  1
   5  1  1  1  0  0  1  1  0  1  0  1  0  1  1  1  1  1  1  0  1  1  1  0  0  1
   6  0  1  1  0  0  0  0  0  0  0  1  0  1  0  0  1  0  0  0  0  1  0  0  0  1
   7  0  1  0  0  0  0  0  0  0  0  0  0  1  0  0  0  0  0  0  0  0  0  0  0  1
   8  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
    Cage
Week 54 55 56 57 58 59 60
   2  1  1  1  1  1  1  1
   3  1  1  1  1  1  1  1
   4  0  1  1  1  1  1  1
   5  0  1  1  1  0  0  0
   6  0  1  0  0  0  0  0
   7  0  0  0  0  0  0  0
   8  0  0  0  0  0  0  0</code></pre>
</div>
<div id="lyer-ptoid-subset" class="section level2">
<h2>LYER-Ptoid subset</h2>
<pre class="r"><code># filter data where BRBR is absent, but LYER and Ptoid survived to time t
LP_df &lt;- filter(timeseries_df, BRBR_Survival == 0 | is.na(BRBR_Survival) == TRUE, LYER_Survival == 1, Mummy_Ptoids_Survival == 1) 

# tabulate variation in timeseries length for each cage
with(LP_df, table(Week, Cage))</code></pre>
<pre><code>    Cage
Week 4 5 6 7 8 9 12 13 14 15 17 18 20 22 23 25 26 27 28 29 30 31 32 33 34 35 36
  4  0 0 0 0 0 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  1
  5  0 0 0 0 0 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  1  1  0  0  1
  6  0 0 0 0 1 0  0  1  0  0  0  0  0  0  0  1  0  0  1  1  0  0  1  1  1  1  1
  7  1 0 1 1 1 1  1  1  0  1  0  1  1  1  1  1  1  1  1  1  0  1  1  1  1  0  1
  8  1 1 1 0 0 1  1  1  1  1  1  1  1  0  1  0  0  0  0  1  1  1  1  1  1  0  1
  9  0 1 1 0 0 1  1  1  1  1  0  1  1  0  1  0  0  0  0  1  1  1  1  1  0  0  1
  10 0 1 1 0 0 1  1  1  0  1  0  1  1  0  1  0  0  0  0  1  1  1  1  0  0  0  1
  11 0 1 1 0 0 1  1  1  0  1  0  1  1  0  1  0  0  0  0  1  1  1  1  0  0  0  1
  12 0 1 1 0 0 1  1  1  0  1  0  1  1  0  1  0  0  0  0  1  1  1  1  0  0  0  1
  13 0 1 1 0 0 1  1  1  0  1  0  1  0  0  1  0  0  0  0  1  1  1  1  0  0  0  1
  14 0 1 1 0 0 1  1  1  0  1  0  1  0  0  1  0  0  0  0  1  0  0  1  0  0  0  1
  15 0 1 1 0 0 0  1  1  0  0  0  1  0  0  1  0  0  0  0  0  0  0  0  0  0  0  0
  16 0 0 0 0 0 0  1  1  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
    Cage
Week 37 38 39 40 41 42 43 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60
  4   0  1  0  1  0  0  0  0  0  1  0  0  0  0  0  0  1  0  0  0  0  0  0
  5   0  1  0  1  0  0  0  0  0  1  0  0  0  1  1  0  1  0  0  0  1  1  1
  6   1  1  0  1  0  1  1  1  1  1  1  0  1  1  1  0  1  0  1  1  1  1  1
  7   1  1  1  1  0  0  1  1  1  1  1  1  1  0  1  0  1  1  0  0  1  1  0
  8   1  1  1  1  1  0  1  1  0  1  1  1  1  0  1  1  1  0  0  0  1  1  0
  9   1  1  0  1  1  0  1  1  0  1  0  1  1  0  1  1  1  0  0  0  1  1  0
  10  1  1  0  1  0  0  1  1  0  1  0  1  1  0  1  1  1  0  0  0  1  1  0
  11  1  1  0  1  0  0  1  1  0  1  0  1  1  0  0  1  1  0  0  0  1  1  0
  12  1  1  0  1  0  0  0  1  0  1  0  1  1  0  0  1  1  0  0  0  1  0  0
  13  1  1  0  1  0  0  0  1  0  1  0  0  1  0  0  1  1  0  0  0  1  0  0
  14  1  0  0  1  0  0  0  1  0  1  0  0  1  0  0  1  1  0  0  0  1  0  0
  15  1  0  0  1  0  0  0  1  0  0  0  0  1  0  0  1  0  0  0  0  1  0  0
  16  1  0  0  0  0  0  0  1  0  0  0  0  1  0  0  1  0  0  0  0  1  0  0</code></pre>
</div>
<div id="lyer-only-subset" class="section level2">
<h2>LYER only subset</h2>
<p>Includes cage-time combinations where only the aphid <em>Lipaphis erysimi</em> persisted.</p>
<pre class="r"><code># filter data so only LYER persists
L_df &lt;- filter(timeseries_df, BRBR_Survival == 0 | is.na(BRBR_Survival) == TRUE, LYER_Survival == 1, Mummy_Ptoids_Survival == 0 | is.na(Mummy_Ptoids_Survival) == TRUE) 

# tabulate variation in timeseries length for each cage
with(L_df, table(Week, Cage))</code></pre>
<pre><code>    Cage
Week 4 5 7 9 14 17 20 22 23 30 31 32 33 34 35 38 39 40 41 43 46 47 48 49 52 59
  7  0 0 0 0  0  0  0  0  0  0  0  0  0  0  1  0  0  0  0  0  0  0  0  0  0  0
  8  0 0 1 0  0  0  0  1  0  0  0  0  0  0  1  0  0  0  0  0  1  0  0  0  0  0
  9  1 0 0 0  0  1  0  1  0  0  0  0  0  1  1  0  1  0  0  0  1  0  1  0  0  0
  10 1 0 0 0  1  1  0  1  0  0  0  0  1  1  1  0  1  0  1  0  1  0  1  0  0  0
  11 1 0 0 0  1  1  0  1  0  0  0  0  1  1  1  0  1  0  1  0  1  0  1  0  1  0
  12 1 0 0 0  1  1  0  1  0  0  0  0  1  1  1  0  1  0  1  1  1  0  1  0  1  1
  13 1 0 0 0  1  1  1  1  0  0  0  0  1  1  1  0  1  0  1  1  1  0  1  1  1  1
  14 1 0 0 0  1  1  1  1  0  1  1  0  1  1  1  1  1  0  1  1  1  0  1  1  1  1
  15 1 0 0 1  1  1  1  1  0  1  1  1  1  1  1  1  1  0  1  1  1  1  1  1  1  1
  16 1 1 0 1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1</code></pre>
</div>
<div id="ptoid-only-subset" class="section level2">
<h2>Ptoid only subset</h2>
<p>Includes cage-time combinations where both aphids went extinct and the parasitoid <em>Diaeretiella rapae</em> survived for 1 to 2 more sampling periods.</p>
<pre class="r"><code># filter data where only ptoid was transiently present in absence of other species
P_df &lt;- filter(timeseries_df, BRBR_Survival == 0 | is.na(BRBR_Survival) == TRUE, LYER_Survival == 0 | is.na(LYER_Survival) == TRUE, Mummy_Ptoids_Survival == 1) 

# tabulate variation in timeseries length for each cage
with(P_df, table(Week, Cage))</code></pre>
<pre><code>    Cage
Week 1 2 3 10 11 15 16 18 19 21 24 25 26 27 28 44 51 54 55 56 57 60
  7  1 0 1  1  1  0  1  0  1  1  0  0  0  0  0  1  1  0  0  1  1  1
  8  0 0 0  0  0  0  1  0  1  0  1  1  1  1  1  0  0  0  1  0  1  0
  9  0 1 0  0  0  0  0  0  0  0  0  1  0  1  0  0  0  0  1  0  0  0
  10 0 1 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
  15 0 0 0  0  0  1  0  0  0  0  0  0  0  0  0  0  0  1  0  0  0  0
  16 0 0 0  0  0  0  0  1  0  0  0  0  0  0  0  0  0  0  0  0  0  0</code></pre>
</div>
</div>
<div id="final-checks" class="section level1">
<h1>Final checks</h1>
<pre class="r"><code># all looks good, no NA or zero values
summary(select(timeseries_df, BRBR_t0, BRBR_t, BRBR_t1)) </code></pre>
<pre><code>      Cage          BRBR_t0           BRBR_t          BRBR_t1      
 Min.   : 1.00   Min.   :  1.00   Min.   :  1.00   Min.   :  1.00  
 1st Qu.:15.75   1st Qu.:  1.00   1st Qu.:  1.00   1st Qu.:  1.00  
 Median :30.50   Median :  1.00   Median :  1.00   Median :  1.00  
 Mean   :30.50   Mean   : 43.74   Mean   : 43.74   Mean   : 43.56  
 3rd Qu.:45.25   3rd Qu.: 21.00   3rd Qu.: 21.00   3rd Qu.: 21.00  
 Max.   :60.00   Max.   :835.00   Max.   :835.00   Max.   :835.00  </code></pre>
<pre class="r"><code>summary(select(timeseries_df, LYER_t0, LYER_t, LYER_t1, LYER_t_adj, LYER_t1_adj)) </code></pre>
<pre><code>      Cage          LYER_t0           LYER_t          LYER_t1      
 Min.   : 1.00   Min.   :   1.0   Min.   :   1.0   Min.   :   1.0  
 1st Qu.:15.75   1st Qu.:   1.0   1st Qu.:   4.0   1st Qu.:   3.0  
 Median :30.50   Median :  10.0   Median :  12.5   Median :  17.5  
 Mean   :30.50   Mean   : 124.5   Mean   : 126.6   Mean   : 131.0  
 3rd Qu.:45.25   3rd Qu.: 155.0   3rd Qu.: 155.0   3rd Qu.: 170.0  
 Max.   :60.00   Max.   :1150.0   Max.   :1150.0   Max.   :1150.0  
   LYER_t_adj      LYER_t1_adj  
 Min.   :   1.0   Min.   :   1  
 1st Qu.:   4.0   1st Qu.:  30  
 Median :  80.0   Median :  92  
 Mean   : 211.9   Mean   : 220  
 3rd Qu.: 301.0   3rd Qu.: 321  
 Max.   :1423.0   Max.   :1423  </code></pre>
<pre class="r"><code>summary(select(timeseries_df, Ptoid_t0, Ptoid_t, Ptoid_t1, Ptoid_t_adj, Ptoid_t1_adj))</code></pre>
<pre><code>      Cage          Ptoid_t0         Ptoid_t          Ptoid_t1     
 Min.   : 1.00   Min.   :  1.00   Min.   :  1.00   Min.   :  1.00  
 1st Qu.:15.75   1st Qu.:  1.00   1st Qu.:  1.00   1st Qu.:  1.00  
 Median :30.50   Median :  1.00   Median :  1.00   Median :  1.00  
 Mean   :30.50   Mean   : 15.34   Mean   : 15.34   Mean   : 15.18  
 3rd Qu.:45.25   3rd Qu.: 11.00   3rd Qu.: 11.00   3rd Qu.: 11.25  
 Max.   :60.00   Max.   :423.00   Max.   :423.00   Max.   :423.00  
  Ptoid_t_adj      Ptoid_t1_adj   
 Min.   :  1.00   Min.   :  1.00  
 1st Qu.:  1.00   1st Qu.:  1.00  
 Median :  5.00   Median :  5.00  
 Mean   : 19.97   Mean   : 19.85  
 3rd Qu.: 19.25   3rd Qu.: 19.25  
 Max.   :454.00   Max.   :454.00  </code></pre>
</div>
<div id="sava-data-for-structural-stability-analyses" class="section level1">
<h1>Sava data for structural stability analyses</h1>
<pre class="r"><code># saved on June 9, 2021
# save(aphids_only_df, full_df, LP_df, L_df, P_df, file = &quot;output/time-series-data.RData&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.7 LTS

Matrix products: default
BLAS:   /usr/lib/libblas/libblas.so.3.6.0
LAPACK: /usr/lib/lapack/liblapack.so.3.6.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] knitr_1.33      cowplot_1.1.1   forcats_0.5.1   stringr_1.4.0  
 [5] dplyr_1.0.6     purrr_0.3.4     readr_1.4.0     tidyr_1.1.3    
 [9] tibble_3.1.2    ggplot2_3.3.3   tidyverse_1.3.1 MASS_7.3-54    
[13] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.6        lattice_0.20-44   lubridate_1.7.10  assertthat_0.2.1 
 [5] rprojroot_2.0.2   digest_0.6.27     utf8_1.2.1        R6_2.5.0         
 [9] cellranger_1.1.0  backports_1.2.1   reprex_2.0.0      evaluate_0.14    
[13] highr_0.9         httr_1.4.2        pillar_1.6.1      rlang_0.4.11     
[17] readxl_1.3.1      rstudioapi_0.13   jquerylib_0.1.4   Matrix_1.3-4     
[21] rmarkdown_2.8     labeling_0.4.2    splines_4.1.0     munsell_0.5.0    
[25] broom_0.7.6       compiler_4.1.0    httpuv_1.6.1      modelr_0.1.8     
[29] xfun_0.23         pkgconfig_2.0.3   mgcv_1.8-36       htmltools_0.5.1.1
[33] tidyselect_1.1.1  fansi_0.5.0       crayon_1.4.1      dbplyr_2.1.1     
[37] withr_2.4.2       later_1.2.0       grid_4.1.0        nlme_3.1-152     
[41] jsonlite_1.7.2    gtable_0.3.0      lifecycle_1.0.0   DBI_1.1.1        
[45] git2r_0.28.0      magrittr_2.0.1    scales_1.1.1      cli_2.5.0        
[49] stringi_1.6.2     farver_2.1.0      fs_1.5.0          promises_1.2.0.1 
[53] xml2_1.3.2        bslib_0.2.5.1     ellipsis_0.3.2    generics_0.1.0   
[57] vctrs_0.3.8       tools_4.1.0       glue_1.4.2        hms_1.1.0        
[61] yaml_2.2.1        colorspace_2.0-1  rvest_1.0.0       haven_2.4.1      
[65] sass_0.4.0       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
