<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Matthew A. Barbour" />

<meta name="date" content="2021-06-24" />

<title>Bayesian MAR(1) models and structural stability</title>

<script src="site_libs/header-attrs-2.8/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">genes-to-foodweb-stability</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mabarbour/genes-to-foodweb-stability">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Bayesian MAR(1) models and structural stability</h1>
<h4 class="author">Matthew A. Barbour</h4>
<h4 class="date">2021-06-24</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-06-24
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>genes-to-foodweb-stability/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200205code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200205)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200205code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200205)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommabarbourgenestofoodwebstabilitytree1955523a9c250979d19c40a7e59a9a41edfee38etargetblank1955523a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mabarbour/genes-to-foodweb-stability/tree/1955523a9c250979d19c40a7e59a9a41edfee38e" target="_blank">1955523</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommabarbourgenestofoodwebstabilitytree1955523a9c250979d19c40a7e59a9a41edfee38etargetblank1955523a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/mabarbour/genes-to-foodweb-stability/tree/1955523a9c250979d19c40a7e59a9a41edfee38e" target="_blank">1955523</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/.Rhistory
    Ignored:    output/.Rapp.history

Untracked files:
    Untracked:  .genes-to-foodweb-stability.Rproj.swp
    Untracked:  analysis/foodweb-persistence.Rmd
    Untracked:  analysis/foodweb-transitions.Rmd
    Untracked:  analysis/plant-growth-no-insects-lme.Rmd
    Untracked:  analysis/prep-timeseries.Rmd
    Untracked:  analysis/structural-stability.Rmd
    Untracked:  code/summarize_stability_metrics.R
    Untracked:  data/InsectAbundanceSurvival.csv
    Untracked:  data/insect_abundance_data_2018-09-26_underneathleaf.csv
    Untracked:  figures/MAR1-parameter-plot.pdf
    Untracked:  figures/aop2-genos-coxadjcurve.pdf
    Untracked:  figures/aop2-genotypes-multistate.pdf
    Untracked:  figures/cage-dynamics-20C.pdf
    Untracked:  figures/cage-dynamics-23C.pdf
    Untracked:  figures/keystone-coxadjcurve.pdf
    Untracked:  figures/keystone-effect-model-comparison-ratio.pdf
    Untracked:  output/all.mar1.brm.adj.rds
    Untracked:  output/all.mar1.brm.unadj.ar2.lag.rds
    Untracked:  output/all.mar1.brm.unadj.noBRBRonLYER.rds
    Untracked:  output/all.mar1.brm.unadj.rds
    Untracked:  output/all.mar1.brm.unadj.xAOP2.rds
    Untracked:  output/initial.mar1.brm.adj.rds
    Untracked:  output/initial.mar1.brm.unadj.rds
    Untracked:  output/time-series-data.RData

Unstaged changes:
    Modified:   README.md
    Deleted:    analysis/about.Rmd
    Deleted:    analysis/community-persistence-keystone.Rmd
    Deleted:    analysis/critical-transitions-keystone.Rmd
    Modified:   analysis/index.Rmd
    Deleted:    analysis/plant-growth-no-insects.Rmd
    Deleted:    analysis/structural-stability-keystone.Rmd
    Deleted:    code/AOP2-LYER-Ptoid-persistence.R
    Modified:   code/README.md
    Deleted:    code/glm-ftest.R
    Modified:   code/plot-feasibility-domain.R
    Deleted:    code/prep-time-series.R
    Modified:   code/simulate-community-dynamics.R
    Deleted:    code/temperature-structural-stability-fig.R
    Modified:   data/README.md
    Deleted:    data/arabidopsis_clean_df.csv
    Modified:   figures/AOP2-growth-no-insects.pdf
    Modified:   figures/MAR1-nonequilibrium-foodchain-AOP2.pdf
    Deleted:    figures/MAR1-nonequilibrium-initial-temp.pdf
    Modified:   figures/MAR1-posterior-foodchain-AOP2.pdf
    Deleted:    figures/MAR1-posterior-initial-temp.pdf
    Deleted:    figures/initial-foodweb-structural-stability.png
    Modified:   figures/keystone-gene.pdf
    Modified:   figures/keystone-structural-stability-forkeynote.pdf
    Modified:   output/README.md
    Deleted:    output/full.mv.norm.brm.keystone.rds
    Deleted:    output/plant-growth-no-insects.RData
    Deleted:    output/reduced.1.brm.keystone.rds
    Deleted:    output/reduced.2.brm.keystone.rds
    Deleted:    output/reduced.3.brm.keystone.rds
    Deleted:    output/reduced.4.brm.keystone.rds
    Deleted:    output/reduced.5.brm.keystone.rds
    Deleted:    output/reduced.6.brm.keystone.rds
    Deleted:    output/timeseries_df.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/structural-stability.Rmd</code>) and HTML (<code>docs/structural-stability.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mabarbour/genes-to-foodweb-stability/blob/399b240b0ca4535da96ca1dbfd1c0d99e7af29cf/analysis/structural-stability.Rmd" target="_blank">399b240</a>
</td>
<td>
mabarbour
</td>
<td>
2020-10-11
</td>
<td>
remove old versions of analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mabarbour/genes-to-foodweb-stability/blob/c1aaf8eef2c50dd6f745e40ac73810b1a1aa5c25/analysis/structural-stability.Rmd" target="_blank">c1aaf8e</a>
</td>
<td>
mabarbour
</td>
<td>
2020-10-11
</td>
<td>
remove old versions of analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mabarbour/genes-to-foodweb-stability/51fc18b769a6ad737e3562079470b87672e5b42e/docs/structural-stability.html" target="_blank">51fc18b</a>
</td>
<td>
mabarbour
</td>
<td>
2020-06-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mabarbour/genes-to-foodweb-stability/blob/86116c87e556ad781ab547037a44f6391ee17703/analysis/structural-stability.Rmd" target="_blank">86116c8</a>
</td>
<td>
mabarbour
</td>
<td>
2020-06-23
</td>
<td>
bioRxiv version of code and data.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mabarbour/genes-to-foodweb-stability/86116c87e556ad781ab547037a44f6391ee17703/docs/structural-stability.html" target="_blank">86116c8</a>
</td>
<td>
mabarbour
</td>
<td>
2020-06-23
</td>
<td>
bioRxiv version of code and data.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="general-priors" class="section level2">
<h2>General priors</h2>
<div id="intrinsic-growth-rates" class="section level3">
<h3>Intrinsic growth rates</h3>
<pre class="r"><code># from Jahan et al. 2014, Journal of Insect Science
# Table 4 lambda (finite rate of increase, discrete time analogue of intrinsic growth rate)
# calculated on a per-day basis and not logged. This is why I multiply by 7 and then take the natural logarithm
Jahan.r.BRBR &lt;- log(c(1.42, 1.36, 1.32, 1.35, 1.40, 1.33, 1.38, 1.37) * 7)
mean(Jahan.r.BRBR) # 2.26</code></pre>
<pre><code>[1] 2.257713</code></pre>
<pre class="r"><code>sd(Jahan.r.BRBR) # 0.02</code></pre>
<pre><code>[1] 0.02468356</code></pre>
<pre class="r"><code># visualize prior
hist(rnorm(1000, mean(Jahan.r.BRBR), sd = 1))</code></pre>
<p><img src="figure/structural-stability.Rmd/r-priors-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-r-priors-1">
Past versions of r-priors-1.png
</button>
</p>
<div id="fig-r-priors-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mabarbour/genes-to-foodweb-stability/blob/86116c87e556ad781ab547037a44f6391ee17703/docs/figure/structural-stability.Rmd/r-priors-1.png" target="_blank">86116c8</a>
</td>
<td>
mabarbour
</td>
<td>
2020-06-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prior.r.BRBR &lt;- &quot;normal(2.26, 1)&quot;

# from Taghizadeh 2019, J. Agr. Sci. Tech.
# Table 2 lambda (finite rate of increase, discrete time analogue of intrinsic growth rate)
# calculated on a per-day basis and not logged. This is why I multiply by 7 and then take the natural logarithm
Tag.r.LYER &lt;- log(c(1.35, 1.30, 1.26, 1.23) * 7)
mean(Tag.r.LYER) # 2.20</code></pre>
<pre><code>[1] 2.196059</code></pre>
<pre class="r"><code>sd(Tag.r.LYER) # 0.04</code></pre>
<pre><code>[1] 0.04028153</code></pre>
<pre class="r"><code># visualize prior
hist(rnorm(1000, mean(Tag.r.LYER), sd = 1))</code></pre>
<p><img src="figure/structural-stability.Rmd/r-priors-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-r-priors-2">
Past versions of r-priors-2.png
</button>
</p>
<div id="fig-r-priors-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mabarbour/genes-to-foodweb-stability/blob/86116c87e556ad781ab547037a44f6391ee17703/docs/figure/structural-stability.Rmd/r-priors-2.png" target="_blank">86116c8</a>
</td>
<td>
mabarbour
</td>
<td>
2020-06-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prior.r.LYER &lt;- &quot;normal(2.20, 1)&quot;

# random effects prior based on variance among cultivars
# I&#39;m just going to use this for all of them, including parasitoids
mean.r.sd &lt;- mean(c(sd(Jahan.r.BRBR), sd(Tag.r.LYER)))
# visualize prior
hist(rnorm(1000, mean = mean.r.sd, sd = 0.5))</code></pre>
<p><img src="figure/structural-stability.Rmd/r-priors-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-r-priors-3">
Past versions of r-priors-3.png
</button>
</p>
<div id="fig-r-priors-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mabarbour/genes-to-foodweb-stability/blob/86116c87e556ad781ab547037a44f6391ee17703/docs/figure/structural-stability.Rmd/r-priors-3.png" target="_blank">86116c8</a>
</td>
<td>
mabarbour
</td>
<td>
2020-06-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># not changing for now
prior.random.effects &lt;- &quot;normal(0.03, 0.5)&quot; # mean of BRBR and LYER

# I don&#39;t have a great sense for the growth rate of the parasitoid, other than that it should be negative
# this seems like a reasonable starting point

# visualize prior
hist(rnorm(1000, mean = -1.5, sd = 1))</code></pre>
<p><img src="figure/structural-stability.Rmd/r-priors-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-r-priors-4">
Past versions of r-priors-4.png
</button>
</p>
<div id="fig-r-priors-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mabarbour/genes-to-foodweb-stability/blob/86116c87e556ad781ab547037a44f6391ee17703/docs/figure/structural-stability.Rmd/r-priors-4.png" target="_blank">86116c8</a>
</td>
<td>
mabarbour
</td>
<td>
2020-06-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prior.r.Ptoid &lt;- &quot;normal(-1.5, 1)&quot;</code></pre>
</div>
<div id="intra--and-interspecific-interactions" class="section level3">
<h3>Intra- and interspecific interactions</h3>
<p>I assume they are weak, i.e. much less than <span class="math inline">\(|1|\)</span>. I also assume that all species exhibit intraspecific competition, aphids have negative interspecific effects with each other, but positive interspecific effects on the parasitoid. I also assume parasitoids have negative interspecific effects on the aphids.</p>
<pre class="r"><code>## intraspecific, 1 = no density dependence. I would prefer to specify an offset first, so that 0 = no density dependence, like the other coefs, but I can&#39;t use the offset if I incorporate measurement error :-(
# visualize prior
hist(rnorm(1000, mean = 0.9, sd = 0.5))</code></pre>
<p><img src="figure/structural-stability.Rmd/interaction-priors-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-interaction-priors-1">
Past versions of interaction-priors-1.png
</button>
</p>
<div id="fig-interaction-priors-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mabarbour/genes-to-foodweb-stability/blob/86116c87e556ad781ab547037a44f6391ee17703/docs/figure/structural-stability.Rmd/interaction-priors-1.png" target="_blank">86116c8</a>
</td>
<td>
mabarbour
</td>
<td>
2020-06-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prior.intra.BRBR &lt;- &quot;normal(0.9, 0.5)&quot;
prior.intra.LYER &lt;- &quot;normal(0.9, 0.5)&quot;
prior.intra.Ptoid &lt;- &quot;normal(0.9, 0.5)&quot;

## negative interspecific, 0 = no interaction
# visualize prior
hist(rnorm(1000, mean = -0.1, sd = 0.5))</code></pre>
<p><img src="figure/structural-stability.Rmd/interaction-priors-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-interaction-priors-2">
Past versions of interaction-priors-2.png
</button>
</p>
<div id="fig-interaction-priors-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mabarbour/genes-to-foodweb-stability/blob/86116c87e556ad781ab547037a44f6391ee17703/docs/figure/structural-stability.Rmd/interaction-priors-2.png" target="_blank">86116c8</a>
</td>
<td>
mabarbour
</td>
<td>
2020-06-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># most of these values are less than 1, which
# is indicative of saturating effects
prior.LYERonBRBR &lt;- &quot;normal(-0.1, 0.5)&quot; 
prior.PtoidonBRBR &lt;- &quot;normal(-0.1, 0.5)&quot;
prior.BRBRonLYER &lt;- &quot;normal(-0.1, 0.5)&quot;
prior.PtoidonLYER &lt;- &quot;normal(-0.1, 0.5)&quot;

## positive interspecific
# visualize prior
hist(rnorm(1000, mean = 0.1, sd = 0.5))</code></pre>
<p><img src="figure/structural-stability.Rmd/interaction-priors-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-interaction-priors-3">
Past versions of interaction-priors-3.png
</button>
</p>
<div id="fig-interaction-priors-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mabarbour/genes-to-foodweb-stability/blob/86116c87e556ad781ab547037a44f6391ee17703/docs/figure/structural-stability.Rmd/interaction-priors-3.png" target="_blank">86116c8</a>
</td>
<td>
mabarbour
</td>
<td>
2020-06-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># most of these values are less than 1, which
# is indicative of saturating effects
prior.BRBRonPtoid &lt;- &quot;normal(0.1, 0.5)&quot;
prior.LYERonPtoid &lt;- &quot;normal(0.1, 0.5)&quot;</code></pre>
</div>
<div id="aop2-effects" class="section level3">
<h3>AOP2 effects</h3>
<p>It was unclear to me <em>a priori</em> exactly how allelic differences at <em>AOP2</em> would affect species’ growth rates or intra- and interspecific interactions. Therefore, I assumed these effects on specific rates could be positive or negative, but would likely be between -1 and 1 (i.e., not ridiculously large).</p>
<pre class="r"><code>prior.AOP2 &lt;- &quot;normal(0, 0.5)&quot;</code></pre>
</div>
<div id="temperature-effects" class="section level3">
<h3>Temperature effects</h3>
<p>As with <em>AOP2</em> it was unclear to me <em>a priori</em> how warming would affect species’ growth rates or intra- and interspecific interactions; therefore, I used the same prior as for <em>AOP2</em>.</p>
<pre class="r"><code>prior.temp &lt;- &quot;normal(0, 0.5)&quot;</code></pre>
</div>
<div id="biomass-effects" class="section level3">
<h3>Biomass effects</h3>
<p>For both aphids, I thought that increasing biomass would increase their intrinsic growth rates, but only weakly, because I didn’t expect biomass to be limiting.</p>
<pre class="r"><code># visualize prior
hist(rnorm(1000, mean = 0.1, sd = 0.5))</code></pre>
<p><img src="figure/structural-stability.Rmd/aphid-biomass-priors-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-aphid-biomass-priors-1">
Past versions of aphid-biomass-priors-1.png
</button>
</p>
<div id="fig-aphid-biomass-priors-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mabarbour/genes-to-foodweb-stability/blob/86116c87e556ad781ab547037a44f6391ee17703/docs/figure/structural-stability.Rmd/aphid-biomass-priors-1.png" target="_blank">86116c8</a>
</td>
<td>
mabarbour
</td>
<td>
2020-06-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prior.AphidBiomass &lt;- &quot;normal(0.1, 0.5)&quot;</code></pre>
<p>For the parasitoid, it was unclear to me whether increasing biomass would have positive or negative effects. I could imagine both, as increasing biomass may increase the search effort of parasitoids, resulting in a negative effect on their growth rate. Alternatively, more biomass may increase the quality of aphids, which could increase the parasitoid’s growth rate. Therefore, I specified a normal prior with mean = 0 and SD = 0.5, so that most of the distribution lied between -1 and 1 (i.e. saturating negative or positive effects).</p>
<pre class="r"><code># visualize prior
hist(rnorm(1000, mean = 0, sd = 0.5))</code></pre>
<p><img src="figure/structural-stability.Rmd/ptoid-biomass-prior-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-ptoid-biomass-prior-1">
Past versions of ptoid-biomass-prior-1.png
</button>
</p>
<div id="fig-ptoid-biomass-prior-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mabarbour/genes-to-foodweb-stability/blob/86116c87e556ad781ab547037a44f6391ee17703/docs/figure/structural-stability.Rmd/ptoid-biomass-prior-1.png" target="_blank">86116c8</a>
</td>
<td>
mabarbour
</td>
<td>
2020-06-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prior.PtoidBiomass &lt;- &quot;normal(0, 0.5)&quot;</code></pre>
</div>
</div>
<div id="model-including-all-species" class="section level2">
<h2>Model including all species</h2>
<div id="formula" class="section level3">
<h3>Formula</h3>
<p>This follows equation 1 in the Supplementary Material.</p>
<pre class="r"><code># BRBR
all.BRBR.bf &lt;- bf(log(BRBR_t1) ~ 0 + Intercept + (me(logBRBR_t, se_logBRBRt) + me(logLYER_t, se_logLYERt) + me(logPtoid_t, se_logPtoidt)) + aop2_genotypes + AOP2_genotypes + temp + log(Biomass_g_t1) + (1|p|Cage))  

# LYER
all.LYER.bf &lt;- bf(log(LYER_t1) ~ 0 + Intercept + (me(logLYER_t, se_logLYERt) + me(logBRBR_t, se_logBRBRt) + me(logPtoid_t, se_logPtoidt)) + aop2_genotypes + AOP2_genotypes + temp + log(Biomass_g_t1) + (1|p|Cage))  

# Ptoid
all.Ptoid.bf &lt;- bf(log(Ptoid_t1) ~ 0 + Intercept + me(logPtoid_t, se_logPtoidt) + me(logBRBR_t, se_logBRBRt) + me(logLYER_t, se_logLYERt) + aop2_genotypes + AOP2_genotypes + temp + log(Biomass_g_t1) + (1|p|Cage))</code></pre>
</div>
<div id="priors-summarized-in-table-s3" class="section level3">
<h3>Priors summarized in Table S3</h3>
<pre class="r"><code>all.mv.priors &lt;- c(
  # aop2 and AOP2 effects
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;aop2_genotypes&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;AOP2_genotypes&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;aop2_genotypes&quot;, resp = &quot;logLYERt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;AOP2_genotypes&quot;, resp = &quot;logLYERt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;aop2_genotypes&quot;, resp = &quot;logPtoidt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;AOP2_genotypes&quot;, resp = &quot;logPtoidt1&quot;),
  # biomass effects
  set_prior(prior.AphidBiomass, class = &quot;b&quot;, coef = &quot;logBiomass_g_t1&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.AphidBiomass, class = &quot;b&quot;, coef = &quot;logBiomass_g_t1&quot;, resp = &quot;logLYERt1&quot;),
  set_prior(prior.PtoidBiomass, class = &quot;b&quot;, coef = &quot;logBiomass_g_t1&quot;, resp = &quot;logPtoidt1&quot;),
  # baseline growth rates
  set_prior(prior.r.BRBR, class = &quot;b&quot;, coef = &quot;Intercept&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.r.LYER, class = &quot;b&quot;, coef = &quot;Intercept&quot;, resp = &quot;logLYERt1&quot;),
  set_prior(prior.r.Ptoid, class = &quot;b&quot;, coef = &quot;Intercept&quot;, resp = &quot;logPtoidt1&quot;),
  # intraspecific effects
  set_prior(prior.intra.BRBR, class = &quot;b&quot;, coef = &quot;melogBRBR_tse_logBRBRt&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.intra.LYER, class = &quot;b&quot;, coef = &quot;melogLYER_tse_logLYERt&quot;, resp = &quot;logLYERt1&quot;),
  set_prior(prior.intra.Ptoid, class = &quot;b&quot;, coef = &quot;melogPtoid_tse_logPtoidt&quot;, resp = &quot;logPtoidt1&quot;),
  # negative interspecific effects
  set_prior(prior.LYERonBRBR, class = &quot;b&quot;, coef = &quot;melogLYER_tse_logLYERt&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.BRBRonLYER, class = &quot;b&quot;, coef = &quot;melogBRBR_tse_logBRBRt&quot;, resp = &quot;logLYERt1&quot;),
  set_prior(prior.PtoidonBRBR, class = &quot;b&quot;, coef = &quot;melogPtoid_tse_logPtoidt&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.PtoidonLYER, class = &quot;b&quot;, coef = &quot;melogPtoid_tse_logPtoidt&quot;, resp = &quot;logLYERt1&quot;),
  # positive interspecific effects
  set_prior(prior.BRBRonPtoid, class = &quot;b&quot;, coef = &quot;melogBRBR_tse_logBRBRt&quot;, resp = &quot;logPtoidt1&quot;),
  set_prior(prior.LYERonPtoid, class = &quot;b&quot;, coef = &quot;melogLYER_tse_logLYERt&quot;, resp = &quot;logPtoidt1&quot;),
  # temp effects
  set_prior(prior.temp, class = &quot;b&quot;, coef = &quot;temp&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.temp, class = &quot;b&quot;, coef = &quot;temp&quot;, resp = &quot;logLYERt1&quot;),
  set_prior(prior.temp, class = &quot;b&quot;, coef = &quot;temp&quot;, resp = &quot;logPtoidt1&quot;),
  # random effects
  set_prior(prior.random.effects, class = &quot;sd&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.random.effects, class = &quot;sd&quot;, resp = &quot;logLYERt1&quot;),
  set_prior(prior.random.effects, class = &quot;sd&quot;, resp = &quot;logPtoidt1&quot;))</code></pre>
</div>
<div id="fit-model" class="section level3">
<h3>Fit Model</h3>
<pre class="r"><code>all.mar1.brm.unadj &lt;- brm(
  data = full_df,
  formula = mvbf(all.BRBR.bf, all.LYER.bf, all.Ptoid.bf) + set_rescor(TRUE),
  iter = 5000,
  save_pars = save_pars(latent = TRUE, all = TRUE),
  prior = all.mv.priors,
  file = &quot;output/all.mar1.brm.unadj.rds&quot;)

all.mar1.brm.unadj</code></pre>
<pre><code> Family: MV(gaussian, gaussian, gaussian) 
  Links: mu = identity; sigma = identity
         mu = identity; sigma = identity
         mu = identity; sigma = identity 
Formula: log(BRBR_t1) ~ 0 + Intercept + (me(logBRBR_t, se_logBRBRt) + me(logLYER_t, se_logLYERt) + me(logPtoid_t, se_logPtoidt)) + aop2_genotypes + AOP2_genotypes + temp + log(Biomass_g_t1) + (1 | p | Cage) 
         log(LYER_t1) ~ 0 + Intercept + (me(logLYER_t, se_logLYERt) + me(logBRBR_t, se_logBRBRt) + me(logPtoid_t, se_logPtoidt)) + aop2_genotypes + AOP2_genotypes + temp + log(Biomass_g_t1) + (1 | p | Cage) 
         log(Ptoid_t1) ~ 0 + Intercept + me(logPtoid_t, se_logPtoidt) + me(logBRBR_t, se_logBRBRt) + me(logLYER_t, se_logLYERt) + aop2_genotypes + AOP2_genotypes + temp + log(Biomass_g_t1) + (1 | p | Cage) 
   Data: full_df (Number of observations: 264) 
Samples: 4 chains, each with iter = 5000; warmup = 2500; thin = 1;
         total post-warmup samples = 10000

Group-Level Effects: 
~Cage (Number of levels: 60) 
                                              Estimate Est.Error l-95% CI
sd(logBRBRt1_Intercept)                           0.18      0.12     0.01
sd(logLYERt1_Intercept)                           0.17      0.10     0.01
sd(logPtoidt1_Intercept)                          0.07      0.06     0.00
cor(logBRBRt1_Intercept,logLYERt1_Intercept)     -0.10      0.48    -0.89
cor(logBRBRt1_Intercept,logPtoidt1_Intercept)     0.08      0.50    -0.85
cor(logLYERt1_Intercept,logPtoidt1_Intercept)     0.01      0.50    -0.88
                                              u-95% CI Rhat Bulk_ESS Tail_ESS
sd(logBRBRt1_Intercept)                           0.46 1.00     2549     4640
sd(logLYERt1_Intercept)                           0.39 1.00     2321     4183
sd(logPtoidt1_Intercept)                          0.21 1.00     6017     4961
cor(logBRBRt1_Intercept,logLYERt1_Intercept)      0.81 1.00     4528     6645
cor(logBRBRt1_Intercept,logPtoidt1_Intercept)     0.91 1.00    12875     7451
cor(logLYERt1_Intercept,logPtoidt1_Intercept)     0.88 1.00    11684     8543

Population-Level Effects: 
                                    Estimate Est.Error l-95% CI u-95% CI Rhat
logBRBRt1_Intercept                     2.15      0.59     0.99     3.31 1.00
logBRBRt1_aop2_genotypes                0.03      0.13    -0.22     0.29 1.00
logBRBRt1_AOP2_genotypes                0.00      0.12    -0.23     0.23 1.00
logBRBRt1_temp                         -0.45      0.08    -0.61    -0.31 1.00
logBRBRt1_logBiomass_g_t1               0.02      0.25    -0.46     0.50 1.00
logLYERt1_Intercept                     3.08      0.51     2.09     4.08 1.00
logLYERt1_aop2_genotypes                0.19      0.11    -0.01     0.40 1.00
logLYERt1_AOP2_genotypes               -0.08      0.10    -0.27     0.11 1.00
logLYERt1_temp                          0.00      0.06    -0.11     0.13 1.00
logLYERt1_logBiomass_g_t1               0.05      0.21    -0.35     0.45 1.00
logPtoidt1_Intercept                   -1.92      0.55    -2.98    -0.84 1.00
logPtoidt1_aop2_genotypes               0.22      0.11     0.01     0.44 1.00
logPtoidt1_AOP2_genotypes              -0.11      0.10    -0.30     0.09 1.00
logPtoidt1_temp                        -0.16      0.06    -0.27    -0.04 1.00
logPtoidt1_logBiomass_g_t1             -1.12      0.22    -1.55    -0.68 1.00
logBRBRt1_melogBRBR_tse_logBRBRt        0.72      0.09     0.54     0.91 1.00
logBRBRt1_melogLYER_tse_logLYERt        0.04      0.12    -0.19     0.27 1.00
logBRBRt1_melogPtoid_tse_logPtoidt     -0.73      0.06    -0.86    -0.61 1.00
logLYERt1_melogLYER_tse_logLYERt        0.34      0.10     0.14     0.54 1.00
logLYERt1_melogBRBR_tse_logBRBRt        0.24      0.08     0.09     0.40 1.00
logLYERt1_melogPtoid_tse_logPtoidt     -0.68      0.05    -0.78    -0.59 1.00
logPtoidt1_melogPtoid_tse_logPtoidt     0.99      0.05     0.88     1.09 1.00
logPtoidt1_melogBRBR_tse_logBRBRt       0.07      0.07    -0.08     0.21 1.00
logPtoidt1_melogLYER_tse_logLYERt       0.45      0.10     0.26     0.64 1.00
                                    Bulk_ESS Tail_ESS
logBRBRt1_Intercept                    10107     7405
logBRBRt1_aop2_genotypes               13547     8138
logBRBRt1_AOP2_genotypes               13124     8587
logBRBRt1_temp                          6602     7763
logBRBRt1_logBiomass_g_t1              11069     8333
logLYERt1_Intercept                     9646     7397
logLYERt1_aop2_genotypes               12154     7630
logLYERt1_AOP2_genotypes               12191     7880
logLYERt1_temp                          7232     6676
logLYERt1_logBiomass_g_t1              11013     8015
logPtoidt1_Intercept                    9133     6787
logPtoidt1_aop2_genotypes              13612     7930
logPtoidt1_AOP2_genotypes              13267     8190
logPtoidt1_temp                        10937     8428
logPtoidt1_logBiomass_g_t1             10685     8894
logBRBRt1_melogBRBR_tse_logBRBRt        6296     6948
logBRBRt1_melogLYER_tse_logLYERt        6039     6329
logBRBRt1_melogPtoid_tse_logPtoidt     11113     8659
logLYERt1_melogLYER_tse_logLYERt        5820     7142
logLYERt1_melogBRBR_tse_logBRBRt        5761     6059
logLYERt1_melogPtoid_tse_logPtoidt     11830     7963
logPtoidt1_melogPtoid_tse_logPtoidt    13008     8341
logPtoidt1_melogBRBR_tse_logBRBRt       9619     7257
logPtoidt1_melogLYER_tse_logLYERt       9713     7690

Family Specific Parameters: 
                 Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sigma_logBRBRt1      1.23      0.06     1.13     1.35 1.00    12496     7477
sigma_logLYERt1      0.95      0.05     0.87     1.05 1.00    10245     7704
sigma_logPtoidt1     1.02      0.05     0.94     1.12 1.00    15968     8374

Residual Correlations: 
                             Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS
rescor(logBRBRt1,logLYERt1)      0.48      0.05     0.38     0.58 1.00     9085
rescor(logBRBRt1,logPtoidt1)    -0.22      0.06    -0.34    -0.10 1.00    14310
rescor(logLYERt1,logPtoidt1)    -0.06      0.06    -0.18     0.07 1.00    13783
                             Tail_ESS
rescor(logBRBRt1,logLYERt1)      8516
rescor(logBRBRt1,logPtoidt1)     8007
rescor(logLYERt1,logPtoidt1)     7635

Samples were drawn using sampling(NUTS). For each parameter, Bulk_ESS
and Tail_ESS are effective sample size measures, and Rhat is the potential
scale reduction factor on split chains (at convergence, Rhat = 1).</code></pre>
</div>
<div id="carrying-capacity-check" class="section level3">
<h3>Carrying capacity check</h3>
<pre class="r"><code># BRBR carrying capacity
fixef(all.mar1.brm.unadj)[&quot;logBRBRt1_Intercept&quot;,&quot;Estimate&quot;] / (1 - fixef(all.mar1.brm.unadj)[&quot;logBRBRt1_melogBRBR_tse_logBRBRt&quot;,&quot;Estimate&quot;]) </code></pre>
<pre><code>[1] 7.81221</code></pre>
<pre class="r"><code>max(log(full_df$BRBR_t1)) # max observed in experiment</code></pre>
<pre><code>[1] 6.727432</code></pre>
<pre class="r"><code># LYER carrying capacity
fixef(all.mar1.brm.unadj)[&quot;logLYERt1_Intercept&quot;,&quot;Estimate&quot;] / (1 - fixef(all.mar1.brm.unadj)[&quot;logLYERt1_melogLYER_tse_logLYERt&quot;,&quot;Estimate&quot;]) # too low</code></pre>
<pre><code>[1] 4.675535</code></pre>
</div>
<div id="reproduce-fig.-s5" class="section level3">
<h3>Reproduce Fig. S5</h3>
<pre class="r"><code># intrinsic growth rates
r_plots &lt;- mcmc_intervals_data(all.mar1.brm.unadj, pars = c(&quot;b_logBRBRt1_Intercept&quot;,&quot;b_logLYERt1_Intercept&quot;,&quot;b_logPtoidt1_Intercept&quot;), prob = 0.66, prob_outer = 0.95) %&gt;%
  mutate(parameter = factor(parameter, levels = c(&quot;b_logPtoidt1_Intercept&quot;,&quot;b_logLYERt1_Intercept&quot;,&quot;b_logBRBRt1_Intercept&quot;))) %&gt;%
  ggplot(aes(x = parameter, y = m)) +
  geom_hline(yintercept = 0, linetype = &quot;dotted&quot;) +
  geom_linerange(aes(ymin = ll, ymax = hh), size = 0.5) +
  geom_linerange(aes(ymin = l, ymax = h), size = 1) + 
  geom_point(size = 1.5) +
  ylab(expression(paste(&quot;Intrinsic growth rates (&quot;, italic(&quot;r&quot;),&quot;)&quot;))) +
  scale_x_discrete(name = &quot;&quot;, labels = c(&quot;D&quot;,&quot;L&quot;,&quot;B&quot;)) + 
  coord_flip() +
  theme_cowplot(font_size = 8)

# AOP2 effect on intrinsic growth rates
delta_r_plots &lt;- mcmc_intervals_data(all.mar1.brm.unadj, regex_pars = c(&quot;_aop2_genotypes&quot;,&quot;_AOP2_genotypes&quot;)) %&gt;%
  separate(col = &quot;parameter&quot;, into = c(&quot;b&quot;,&quot;species&quot;,&quot;allele&quot;,&quot;genotypes&quot;)) %&gt;%
  mutate(species = factor(species, levels = c(&quot;logPtoidt1&quot;,&quot;logLYERt1&quot;,&quot;logBRBRt1&quot;))) %&gt;%
  ggplot(aes(x = species, y = m, color = allele)) +
  geom_hline(yintercept = 0, linetype = &quot;dotted&quot;, color = &quot;grey&quot;) +
  geom_point(position = position_dodge(width = 0.4), size = 1.5) +
  geom_linerange(aes(ymin = l, ymax = h), size = 1, position = position_dodge(width = 0.4)) + 
  geom_linerange(aes(ymin = ll, ymax = hh), size = 0.5, position = position_dodge(width = 0.4)) +
  scale_color_grey(name = &quot;&quot;, labels = c(&quot;AOP2\u2013&quot;, &quot;AOP2+&quot;)) + # viridis_d
  ylab(expression(paste(&quot;Allele effect (&quot;, Delta*&quot;r&quot;, italic(&quot;G&quot;),&quot;)&quot;))) +
  scale_x_discrete(name = &quot;&quot;, labels = c(&quot;D&quot;,&quot;L&quot;,&quot;B&quot;)) + #c(expression(italic(&quot;D. rapae&quot;)), expression(italic(&quot;L. erysimi&quot;)), expression(italic(&quot;B. brassicae&quot;)))) +
  theme_cowplot(font_size = 8) +
  coord_flip()

# Interaction matrix
b_data &lt;- mcmc_intervals_data(
  all.mar1.brm.unadj, 
   pars = c(&quot;bsp_logBRBRt1_melogBRBR_tse_logBRBRt&quot;,
            &quot;bsp_logBRBRt1_melogLYER_tse_logLYERt&quot;,
            &quot;bsp_logBRBRt1_melogPtoid_tse_logPtoidt&quot;,
            &quot;bsp_logLYERt1_melogBRBR_tse_logBRBRt&quot;,
            &quot;bsp_logLYERt1_melogLYER_tse_logLYERt&quot;,
            &quot;bsp_logLYERt1_melogPtoid_tse_logPtoidt&quot;,
            &quot;bsp_logPtoidt1_melogBRBR_tse_logBRBRt&quot;,
            &quot;bsp_logPtoidt1_melogLYER_tse_logLYERt&quot;,
            &quot;bsp_logPtoidt1_melogPtoid_tse_logPtoidt&quot;),
  prob = 0.66, prob_outer = 0.95) %&gt;%
  mutate(parameter = factor(parameter, 
                            labels = c(&quot;B \u2b62 B&quot;,&quot;L \u2b62 B&quot;,&quot;D \u2b62 B&quot;,&quot;B \u2b62 L&quot;,&quot;L \u2b62 L&quot;,&quot;D \u2b62 L&quot;,&quot;B \u2b62 D&quot;,&quot;L \u2b62 D&quot;,&quot;D \u2b62 D&quot;)),
         baseline = c(1,0,0,
                      0,1,0,
                      0,0,1))

b_plots &lt;- ggplot(b_data, aes(x = parameter, y = m, color = factor(baseline))) +
  geom_hline(aes(yintercept = baseline, color = factor(baseline)), linetype = &quot;dotted&quot;)  +
  geom_linerange(aes(ymin = ll, ymax = hh), size = 0.5) +
  geom_linerange(aes(ymin = l, ymax = h), size = 1) + 
  geom_point(size = 1.5) +
  ylab(expression(paste(&quot;Interactions (&quot;, italic(&quot;b&quot;),&quot;)&quot;))) +
  scale_x_discrete(name = &quot;&quot;, limits = rev) + 
  theme_cowplot(font_size = 8) +
  coord_flip() + 
  scale_color_grey(name = &quot;&quot;, labels = c(&quot;Inter&quot;,&quot;Intra&quot;)) # viridis_d

# biomass effect on intrinsic growth rates
p_plots &lt;- mcmc_intervals_data(all.mar1.brm.unadj, regex_pars = &quot;Biomass_g_t1$&quot;, prob = 0.66, prob_outer = 0.95) %&gt;%
  mutate(parameter = factor(parameter, levels = c(&quot;b_logPtoidt1_logBiomass_g_t1&quot;,&quot;b_logLYERt1_logBiomass_g_t1&quot;,&quot;b_logBRBRt1_logBiomass_g_t1&quot;))) %&gt;%
  ggplot(aes(x = parameter, y = m)) +
  geom_hline(yintercept = 0, linetype = &quot;dotted&quot;) +
  geom_linerange(aes(ymin = ll, ymax = hh), size = 0.5) +
  geom_linerange(aes(ymin = l, ymax = h), size = 1) + 
  geom_point(size = 1.5) +
  ylab(expression(paste(&quot;Plant biomass effect (&quot;, italic(&quot;p&quot;),&quot;)&quot;))) +
  scale_x_discrete(name = &quot;&quot;, labels = c(&quot;D&quot;,&quot;L&quot;,&quot;B&quot;)) +
  coord_flip() +
  theme_cowplot(font_size = 8)

# temperature effect on intrinsic growth rates
t_plots &lt;- mcmc_intervals_data(all.mar1.brm.unadj, regex_pars = &quot;temp$&quot;, prob = 0.66, prob_outer = 0.95) %&gt;%
  mutate(parameter = factor(parameter, levels = c(&quot;b_logPtoidt1_temp&quot;,&quot;b_logLYERt1_temp&quot;,&quot;b_logBRBRt1_temp&quot;))) %&gt;%
  ggplot(aes(x = parameter, y = m)) +
  geom_hline(yintercept = 0, linetype = &quot;dotted&quot;) +
  geom_linerange(aes(ymin = ll, ymax = hh), size = 0.5) +
  geom_linerange(aes(ymin = l, ymax = h), size = 1) + 
  geom_point(size = 1.5) +
  ylab(expression(paste(&quot;Temperature effect (&quot;, Delta*&quot;r&quot;, italic(&quot;T&quot;),&quot;)&quot;))) +
  scale_x_discrete(name = &quot;&quot;, labels = c(&quot;D&quot;,&quot;L&quot;,&quot;B&quot;)) +
  coord_flip() +
  theme_cowplot(font_size = 8)

# combine plots
MAR1_parameter_plot &lt;- plot_grid(r_plots, delta_r_plots, t_plots, b_plots, p_plots, nrow = 3, ncol = 2, align = &#39;v&#39;, axis = &#39;l&#39;, rel_widths = c(0.8,1))
x11(); MAR1_parameter_plot
#save_plot(filename = &quot;figures/MAR1-parameter-plot.pdf&quot;, plot = MAR1_parameter_plot, device=cairo_pdf, base_asp = 1.1)</code></pre>
<p><img src="figure/structural-stability.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="structural-stability" class="section level3">
<h3>Structural stability</h3>
<pre class="r"><code>stability_all.mar1.brm.unadj &lt;- aop2_vs_AOP2_posterior_samples_unadj(all.mar1.brm.unadj, n.geno = 2, temp.value = 0, logbiomass.value = 0) 
stability_all.mar1.brm.unadj$aop2_SS_LP_BayesP # clearly stabilizing</code></pre>
<pre><code>[1] 0.9966</code></pre>
<pre class="r"><code>stability_all.mar1.brm.unadj$aop2_SS_LP_effect</code></pre>
<pre><code>[1] 13.0206</code></pre>
<pre class="r"><code>stability_all.mar1.brm.unadj$aop2_SS_LP_95CI</code></pre>
<pre><code>     2.5%     97.5% 
 3.873221 23.478849 </code></pre>
</div>
<div id="aphid-and-parasitoid-intrinsic-growth-rates" class="section level3">
<h3>Aphid and parasitoid intrinsic growth rates</h3>
<pre class="r"><code>stability_all.mar1.brm.unadj$aop2_r_LYER_effect</code></pre>
<pre><code>[1] 0.5389944</code></pre>
<pre class="r"><code>stability_all.mar1.brm.unadj$aop2_r_LYER_95CI</code></pre>
<pre><code>       2.5%       97.5% 
-0.04038828  1.12561843 </code></pre>
<pre class="r"><code>stability_all.mar1.brm.unadj$aop2_r_Ptoid_effect</code></pre>
<pre><code>[1] 0.6588181</code></pre>
<pre class="r"><code>stability_all.mar1.brm.unadj$aop2_r_Ptoid_95CI</code></pre>
<pre><code>     2.5%     97.5% 
0.0859789 1.2534024 </code></pre>
<pre class="r"><code>stability_all.mar1.brm.unadj_n.geno1 &lt;- aop2_vs_AOP2_posterior_samples_unadj(all.mar1.brm.unadj, n.geno = 1, temp.value = 0, logbiomass.value = 0)
stability_all.mar1.brm.unadj_n.geno1$aop2_r_LYER_effect</code></pre>
<pre><code>[1] 0.2694972</code></pre>
<pre class="r"><code>stability_all.mar1.brm.unadj_n.geno1$aop2_r_LYER_95CI</code></pre>
<pre><code>       2.5%       97.5% 
-0.02019414  0.56280922 </code></pre>
</div>
<div id="reproduce-fig.-s6-and-s7" class="section level3">
<h3>Reproduce Fig. S6 and S7</h3>
<pre class="r"><code># merge full data set and that with remaining 
complete_df &lt;- bind_rows(aphids_only_df, full_df, LP_df, L_df, P_df) %&gt;% 
  arrange(Cage, Week)

predict_data &lt;- complete_df

# get predictions
BRBR_predict_df &lt;- predict_data %&gt;% filter(BRBR_Survival == 1) %&gt;% mutate(Abundance = BRBR_t1)
predict_BRBRt1 &lt;- bind_cols(BRBR_predict_df, data.frame(predict(all.mar1.brm.unadj, newdata = BRBR_predict_df, resp = &quot;logBRBRt1&quot;))) %&gt;% mutate(Species = &quot;BRBR&quot;) 

LYER_predict_df &lt;- predict_data %&gt;% filter(LYER_Survival == 1) %&gt;% mutate(Abundance = LYER_t1)
predict_LYERt1 &lt;- bind_cols(LYER_predict_df, data.frame(predict(all.mar1.brm.unadj, newdata = LYER_predict_df, resp = &quot;logLYERt1&quot;))) %&gt;% mutate(Species = &quot;LYER&quot;) 

Ptoid_predict_df &lt;- predict_data %&gt;% filter(Mummy_Ptoids_Survival == 1) %&gt;% mutate(Abundance = Ptoid_t1)
predict_Ptoidt1 &lt;- bind_cols(Ptoid_predict_df, data.frame(predict(all.mar1.brm.unadj, newdata = Ptoid_predict_df, resp = &quot;logPtoidt1&quot;))) %&gt;% mutate(Species = &quot;Ptoid&quot;) 

# combine data
predict_all &lt;- bind_rows(predict_BRBRt1, predict_LYERt1, predict_Ptoidt1) %&gt;%
  mutate(Species = factor(Species, levels = c(&quot;BRBR&quot;,&quot;LYER&quot;,&quot;Ptoid&quot;), labels = c(&quot;B. brassicae&quot;, &quot;L. erysimi&quot;, &quot;D. rapae&quot;)))

# get week when first species went extinct (always BRBR at least)
full_df_last_week &lt;- full_df %&gt;%
  group_by(Cage) %&gt;%
  summarise(last_week = last(Week)) 

# would be nice to color code facets by number of aop2 genotypes e.g.
# something like? https://stackoverflow.com/questions/19440069/ggplot2-facet-wrap-strip-color-based-on-variable-in-data-set

# 20 C cages
plot_cage_dynamics20C &lt;- filter(predict_all, Cage %in% 1:30) %&gt;% 
  left_join(., full_df_last_week) %&gt;%
  ggplot(aes(x = Week, y = log(Abundance), group = interaction(Cage, Species))) +
  geom_ribbon(aes(ymin = Q2.5, ymax = Q97.5, fill = Species), alpha = 0.25) +
  geom_line(aes(y = Estimate, color = Species)) +
  geom_jitter(aes(color = Species), size = 0.5) + 
  facet_wrap(~Cage, nrow = 5, ncol = 6) +
  scale_color_viridis_d() +
  scale_fill_viridis_d() +
  theme_cowplot(font_size = 10) +
  geom_vline(xintercept = 1.5, linetype = &quot;dotted&quot;, color = &quot;grey&quot;) +
  geom_vline(aes(xintercept = last_week), linetype = &quot;dotted&quot;, color = &quot;grey&quot;) +
  coord_cartesian(ylim = c(0,8)) +
  ylab(expression(log(Abundance[t+1])))
x11(); plot_cage_dynamics20C
#ggsave(plot = plot_cage_dynamics20C, filename = &quot;figures/cage-dynamics-20C.pdf&quot;, height = 5, width = 6, device=cairo_pdf)

# 23 C cages
plot_cage_dynamics23C &lt;- filter(predict_all, Cage %in% 31:60) %&gt;%
  left_join(., full_df_last_week) %&gt;%
  ggplot(aes(x = Week, y = log(Abundance), group = interaction(Cage, Species))) +
  geom_ribbon(aes(ymin = Q2.5, ymax = Q97.5, fill = Species), alpha = 0.25) +
  geom_line(aes(y = Estimate, color = Species)) +
  geom_point(aes(color = Species), size = 0.5) +
  facet_wrap(~Cage, nrow = 5, ncol = 6) +
  scale_color_viridis_d() +
  scale_fill_viridis_d() +
  theme_cowplot(font_size = 10) + 
  geom_vline(xintercept = 1.5, linetype = &quot;dotted&quot;, color = &quot;grey&quot;) +
  geom_vline(aes(xintercept = last_week), linetype = &quot;dotted&quot;, color = &quot;grey&quot;) +
  coord_cartesian(ylim = c(0,8)) +
  ylab(expression(log(Abundance[t+1])))</code></pre>
<p><img src="figure/structural-stability.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>x11(); plot_cage_dynamics23C
#ggsave(plot = plot_cage_dynamics23C, filename = &quot;figures/cage-dynamics-23C.pdf&quot;, height = 5, width = 6, device=cairo_pdf)</code></pre>
<p><img src="figure/structural-stability.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="mar1-bayesian-r2-in-table-s4" class="section level3">
<h3>MAR(1) Bayesian R<sup>2</sup> in Table S4</h3>
<pre class="r"><code>bayes_R2(all.mar1.brm.unadj, newdata = BRBR_predict_df, resp = &quot;logBRBRt1&quot;)</code></pre>
<pre><code>             Estimate  Est.Error      Q2.5     Q97.5
R2logBRBRt1 0.6458531 0.02399933 0.5962244 0.6880834</code></pre>
<pre class="r"><code>bayes_R2(all.mar1.brm.unadj, newdata = LYER_predict_df, resp = &quot;logLYERt1&quot;)</code></pre>
<pre><code>             Estimate  Est.Error      Q2.5     Q97.5
R2logLYERt1 0.4869181 0.03248254 0.4240039 0.5503659</code></pre>
<pre class="r"><code>bayes_R2(all.mar1.brm.unadj, newdata = Ptoid_predict_df, resp = &quot;logPtoidt1&quot;)</code></pre>
<pre><code>              Estimate  Est.Error      Q2.5     Q97.5
R2logPtoidt1 0.6289082 0.01183945 0.6005009 0.6462458</code></pre>
</div>
<div id="reproduce-core-of-fig.-4" class="section level3">
<h3>Reproduce core of Fig. 4</h3>
<p>Plot the effect of <em>AOP2</em> gene on structural stability of three-species food chain.</p>
<pre class="r"><code>median_all.mar1.brm.unadj &lt;- aop2_vs_AOP2_median_effects_unadj(all.mar1.brm.unadj, n.geno = 2, temp.value = 0, logbiomass.value = 0)

# get raw data for manually making plot
get_FD.2sp &lt;- FeasibilityDomain2sp(A = list(median_all.mar1.brm.unadj$aop2.mat[2:3,2:3], 
                                            median_all.mar1.brm.unadj$AOP2.mat[2:3,2:3]),
                     r = list(median_all.mar1.brm.unadj$aop2.IGR[2:3], 
                              median_all.mar1.brm.unadj$AOP2.IGR[2:3]),
                     labels = c(&quot;aop2&quot;, &quot;AOP2&quot;),
                     normalize = TRUE) %&gt;%
  rename(aop2 = A_ID) 

# Draw polygon for feasibility domain
# from: https://stackoverflow.com/questions/12794596/how-fill-part-of-a-circle-using-ggplot2
# define the circle; add a point at the center if the &#39;pie slice&#39; if the shape is to be filled
circleFun &lt;- function(center=c(0,0), diameter=1, npoints=100, start=0, end=2, filled=TRUE){
  tt &lt;- seq(start*pi, end*pi, length.out=npoints)
  df &lt;- data.frame(
    x = center[1] + diameter / 2 * cos(tt),
    y = center[2] + diameter / 2 * sin(tt)
  )
  if(filled==TRUE) { #add a point at the center so the whole &#39;pie slice&#39; is filled
    df &lt;- rbind(df, center)
  }
  return(df)
}

## plot figure 4
# alpha_level &lt;- 0.05 # very low so use better looking arrows with keynote
plot_fig_4 &lt;- ggplot(filter(get_FD.2sp, Type == &quot;r&quot;), aes(x = Sp_1, y = Sp_2)) + 
  # draw intrinsic growth rate vectors
  geom_segment(aes(x = 0, y = 0, xend = Sp_1, yend = Sp_2, linetype = aop2), # alpha for manuscript
               arrow = arrow(type = &#39;open&#39;, length = unit(0.1,&quot;cm&quot;))) +
  # draw critical boundary (remove for final plot after adjusting geom_polygon)
  # geom_segment(data = filter(get_FD.2sp, Type == &quot;A&quot;)[c(1,3),], # just need one lower bound
  #              aes(x = 0, y = 0, xend = Sp_1, yend = Sp_2, alpha = aop2),
  #              linetype = &quot;solid&quot;,
  #              size = 0.5) +
  xlab(&quot;Aphid growth rate (normalized)&quot;) + 
  ylab(&quot;Parasitoid growth rate (normalized)&quot;) +
  # illustrate circular nature of feasibility domain
  coord_cartesian(xlim = c(0,1), ylim = c(-0.9,0), expand = F) + 
  # scale_alpha_manual(values = c(alpha_level, alpha_level), labels = c(&quot;AOP2\u2013&quot;,&quot;AOP2+&quot;), name = &quot;&quot;) + # used for manuscript plot to provide better looking arrows with keynote
  scale_linetype_manual(values = c(&quot;solid&quot;, &quot;dashed&quot;), labels = c(&quot;AOP2\u2013&quot;,&quot;AOP2+&quot;), name = &quot;&quot;) +
  # adjusted until critical boundary was correct, then removed critical boundary for final plot
  geom_polygon(data=circleFun(c(0,0), diameter=2, start=0, end=-0.192, npoints=100, filled=TRUE), 
               aes(x,y), alpha = 0.1, inherit.aes = F) +
  theme_cowplot(font_size = 18, line_size = 1)
x11(); plot_fig_4
# changes for final version
# alpha = aop2 in geom_segment(aes())
# uncomment scale_alpha_manual
# comment out scale_linetype_manual
# remove line for critical boundary
# alpha version saved for keynote # ggsave(plot = plot_fig_4, filename = &quot;figures/keystone-structural-stability-forkeynote.pdf&quot;, width = 8, height = 8, units = &quot;in&quot;)</code></pre>
<p><img src="figure/structural-stability.Rmd/aop2-feasibility-domain-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>I then used Keynote to manually add images and text to create the final version presented in Figure 4 of the main text.</p>
</div>
<div id="reproduce-fig.-s8" class="section level3">
<h3>Reproduce Fig. S8</h3>
<pre class="r"><code># subsample 1/8 of the posterior to make it easier to visualize
rsamp &lt;- sample(1:10000, size = 1000)

# plot
plot_MAR1_posterior_foodchain_AOP2 &lt;- stability_all.mar1.brm.unadj$all.aop2_vs_AOP2_stability.df %&gt;%
  #filter(r_Ptoid &lt; 0) %&gt;%
  filter(posterior_sample %in% rsamp) %&gt;%
  mutate(n.allele = as.numeric(as.factor(aop2_vs_AOP2))) %&gt;%
  ggplot(., aes(x = n.allele, y = FeasibilityBoundaryLYER.Ptoid)) +
  geom_line(aes(group = posterior_sample), alpha = 0.1) +
  stat_summary(fun.y = mean, geom = &quot;line&quot;, color = &quot;firebrick1&quot;, size = 1) +
  stat_summary(fun.y = mean, geom = &quot;point&quot;, color = &quot;firebrick1&quot;, size = 1.5) +
  theme_minimal_hgrid() +
  scale_x_continuous(name = &quot;Allele&quot;, breaks = c(1,2), labels = c(&quot;AOP2+&quot;,&quot;AOP2\u2013&quot;), expand = c(0.1,0.1)) +
  ylab(&quot;Normalized angle from critical boundary&quot;) 
#ggsave(filename = &quot;figures/MAR1-posterior-foodchain-AOP2.pdf&quot;, width = 6, height = 5, device=cairo_pdf)</code></pre>
</div>
<div id="reproduce-fig.-s9" class="section level3">
<h3>Reproduce Fig. S9</h3>
<p>The above plot illustrates the effect of <em>AOP2</em> on the structural stability of the equilibrium abundances of species. I can explore whether our results hold in a non-equilibrium scenario that better characterizes our observational data.</p>
<p>To do this, I look at the the effect of <em>AOP2</em> across a range of initial conditions for the abundances of LYER and Ptoid. I get this data by simulating community dynamics with the observed effects of <em>AOP2</em> across a range of initial conditions. I restricted our simulation to 10 time steps, as BRBR went extinct commonly at week 7 (experiment was 17 weeks long). I also set an extinction threshold of 2 individuals.</p>
<pre class="r"><code>LP_duration &lt;- 10
LP_threshold &lt;- log(2) # set threshold of two individuals in the populations
res &lt;- 0.1
LP_test_df &lt;- expand.grid(LYER = seq(1, 6, by = res), Ptoid = seq(1, 6, by = res))

## simulate population dynamics and determine which species goes extinct

# aop2
FE_LP_aop2 &lt;- list()
for(i in 1:length(LP_test_df$LYER)){
  FE_LP_aop2[[i]] &lt;- first_extinction_2sp(Initial.States = c(LYER = LP_test_df[i,&quot;LYER&quot;], Ptoid = LP_test_df[i,&quot;Ptoid&quot;]), 
                                          Interaction.Matrix = median_all.mar1.brm.unadj$aop2.mat[2:3,2:3] + diag(2), 
                                          IGR.Vector = median_all.mar1.brm.unadj$aop2.IGR[2:3], 
                                          Duration = LP_duration, 
                                          threshold = LP_threshold)
}
FE_LP_aop2_df &lt;- bind_cols(LP_test_df, plyr::ldply(FE_LP_aop2)) %&gt;%
  mutate(allele = &quot;aop2&quot;)

# AOP2
FE_LP_AOP2 &lt;- list()
for(i in 1:length(LP_test_df$LYER)){
  FE_LP_AOP2[[i]] &lt;- first_extinction_2sp(Initial.States = c(LYER = LP_test_df[i,&quot;LYER&quot;], Ptoid = LP_test_df[i,&quot;Ptoid&quot;]), 
                                           Interaction.Matrix = median_all.mar1.brm.unadj$AOP2.mat[2:3,2:3] + diag(2), 
                                           IGR.Vector = median_all.mar1.brm.unadj$AOP2.IGR[2:3], 
                                           Duration = LP_duration, 
                                           threshold = LP_threshold)
}
FE_LP_AOP2_df &lt;- bind_cols(LP_test_df, plyr::ldply(FE_LP_AOP2)) %&gt;%
  mutate(allele = &quot;AOP2&quot;)

# get observed data on initial abundances of LYER and Ptoid after BRBR went extinct
cage_type &lt;- LP_df %&gt;%
  distinct(Cage, aop2_vs_AOP2) %&gt;%
  mutate(allele = ifelse(aop2_vs_AOP2 &gt; 0, 1,
                         ifelse(aop2_vs_AOP2 &lt; 0, -1, NA)))

LP_actual_df &lt;- LP_df %&gt;%
  group_by(Cage) %&gt;% # rich,
  summarise_at(vars(LYER_t, Ptoid_t), list(first = first)) %&gt;%
  ungroup() %&gt;%
  mutate(log_LYER_t_first = log(LYER_t_first),
         log_Ptoid_t_first = log(Ptoid_t_first)) %&gt;%
  as.data.frame() %&gt;%
  left_join(., cage_type) %&gt;%
  drop_na() %&gt;% # don&#39;t use this in order to plot all cages
  mutate(allele = factor(allele, levels = c(1,-1), labels = c(&quot;AOP2\u2013&quot;,&quot;AOP2+&quot;)))</code></pre>
<p>The graph below shows a couple of useful things. First, our predictions hold for outside of equilibrium. That is, there is a greater likelihood of LYER-Ptoid persistence when there are genotypes with the null <em>AOP2</em><span class="math inline">\(-\)</span> allele in the population.</p>
<p>It’s also important to note that there is a region of parameter space where LYER goes extinct before Ptoid, which would eventually lead to the collapse of the Ptoid since it has lost its resource. This is not possible if I were to assume the community is at equilibrium.</p>
<pre class="r"><code>cbPalette &lt;-  viridis::viridis(4)

with(bind_rows(FE_LP_aop2_df, FE_LP_AOP2_df), table(species))</code></pre>
<pre><code>species
 LYER Ptoid 
 1681  1766 </code></pre>
<pre class="r"><code>plot_fig_S9 &lt;- bind_rows(FE_LP_aop2_df, FE_LP_AOP2_df) %&gt;%
  mutate(allele = factor(allele, labels = c(&quot;AOP2\u2013&quot;,&quot;AOP2+&quot;))) %&gt;%
  mutate(species = ifelse(is.na(species) == T, &quot;Food chain persists&quot;, species),
         fspecies = factor(species, levels = c(&quot;LYER&quot;,&quot;Ptoid&quot;,&quot;Food chain persists&quot;), labels = c(&quot;Arabidopsis only&quot;,&quot;Aphid only&quot;,&quot;Food chain persists&quot;))) %&gt;%
  ggplot(., aes(x = LYER, y = Ptoid)) + # fspecies
  geom_tile(aes(fill = fspecies)) +
  #geom_point(data = LP_actual_df, aes(x = log_LYER_t_adj_first, y = log_Ptoid_t_adj_first)) + 
  facet_grid(~allele) +
  #scale_fill_viridis_d(name = &quot;Critical transition&quot;) +
  scale_fill_manual(name = &quot;Food-web transition&quot;, values = cbPalette[1:3]) + 
  coord_cartesian(xlim = c(1, max(LP_actual_df$log_LYER_t_first)), 
                  ylim = c(1, max(LP_actual_df$log_Ptoid_t_first))) +
  xlab(&quot;Aphid initial abundance (log scale)&quot;) +
  ylab(&quot;Parasitoid initial abundance (log scale)&quot;) +
  theme_cowplot() +
  theme(strip.background = element_blank()) 
x11(); plot_fig_S9
#ggsave(plot = plot_fig_S9, filename = &quot;figures/MAR1-nonequilibrium-foodchain-AOP2.pdf&quot;, width = 6, height = 5, device=cairo_pdf)</code></pre>
<p><img src="figure/structural-stability.Rmd/nonequilibrium-plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-nonequilibrium-plot-1">
Past versions of nonequilibrium-plot-1.png
</button>
</p>
<div id="fig-nonequilibrium-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mabarbour/genes-to-foodweb-stability/blob/86116c87e556ad781ab547037a44f6391ee17703/docs/figure/structural-stability.Rmd/nonequilibrium-plot-1.png" target="_blank">86116c8</a>
</td>
<td>
mabarbour
</td>
<td>
2020-06-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="alternative-models" class="section level2">
<h2>Alternative models</h2>
<div id="drop-brbr---lyer" class="section level3">
<h3>Drop BRBR -&gt; LYER</h3>
<pre class="r"><code># update LYER formula
all.LYER.bf.noBRBRonLYER &lt;- update(all.LYER.bf, .~. -me(logBRBR_t, se_logBRBRt))

all.mar1.brm.unadj.noBRBRonLYER &lt;- brm(
  data = full_df,
  formula = mvbf(all.BRBR.bf, all.LYER.bf.noBRBRonLYER, all.Ptoid.bf) + set_rescor(TRUE), 
  iter = 5000,
  save_pars = save_pars(latent = TRUE, all = TRUE),
  prior = all.mv.priors[-17,],
  file = &quot;output/all.mar1.brm.unadj.noBRBRonLYER.rds&quot;)

all.mar1.brm.unadj.noBRBRonLYER</code></pre>
<pre><code> Family: MV(gaussian, gaussian, gaussian) 
  Links: mu = identity; sigma = identity
         mu = identity; sigma = identity
         mu = identity; sigma = identity 
Formula: log(BRBR_t1) ~ 0 + Intercept + (me(logBRBR_t, se_logBRBRt) + me(logLYER_t, se_logLYERt) + me(logPtoid_t, se_logPtoidt)) + aop2_genotypes + AOP2_genotypes + temp + log(Biomass_g_t1) + (1 | p | Cage) 
         log(LYER_t1) ~ Intercept + me(logLYER_t, se_logLYERt) + me(logPtoid_t, se_logPtoidt) + aop2_genotypes + AOP2_genotypes + temp + log(Biomass_g_t1) + (1 | p | Cage) - 1 
         log(Ptoid_t1) ~ 0 + Intercept + me(logPtoid_t, se_logPtoidt) + me(logBRBR_t, se_logBRBRt) + me(logLYER_t, se_logLYERt) + aop2_genotypes + AOP2_genotypes + temp + log(Biomass_g_t1) + (1 | p | Cage) 
   Data: full_df (Number of observations: 264) 
Samples: 4 chains, each with iter = 5000; warmup = 2500; thin = 1;
         total post-warmup samples = 10000

Group-Level Effects: 
~Cage (Number of levels: 60) 
                                              Estimate Est.Error l-95% CI
sd(logBRBRt1_Intercept)                           0.19      0.13     0.01
sd(logLYERt1_Intercept)                           0.09      0.07     0.00
sd(logPtoidt1_Intercept)                          0.08      0.06     0.00
cor(logBRBRt1_Intercept,logLYERt1_Intercept)      0.04      0.49    -0.86
cor(logBRBRt1_Intercept,logPtoidt1_Intercept)     0.08      0.49    -0.85
cor(logLYERt1_Intercept,logPtoidt1_Intercept)     0.03      0.50    -0.87
                                              u-95% CI Rhat Bulk_ESS Tail_ESS
sd(logBRBRt1_Intercept)                           0.48 1.00     1726     3382
sd(logLYERt1_Intercept)                           0.25 1.00     3642     4521
sd(logPtoidt1_Intercept)                          0.21 1.00     5652     5130
cor(logBRBRt1_Intercept,logLYERt1_Intercept)      0.88 1.00     7181     6805
cor(logBRBRt1_Intercept,logPtoidt1_Intercept)     0.90 1.00     8647     6516
cor(logLYERt1_Intercept,logPtoidt1_Intercept)     0.88 1.00     7696     7899

Population-Level Effects: 
                                    Estimate Est.Error l-95% CI u-95% CI Rhat
logBRBRt1_Intercept                     2.33      0.59     1.18     3.48 1.00
logBRBRt1_aop2_genotypes                0.04      0.13    -0.22     0.30 1.00
logBRBRt1_AOP2_genotypes                0.01      0.12    -0.23     0.24 1.00
logBRBRt1_temp                         -0.52      0.08    -0.67    -0.37 1.00
logBRBRt1_logBiomass_g_t1              -0.02      0.25    -0.50     0.46 1.00
logLYERt1_Intercept                     3.49      0.49     2.52     4.45 1.00
logLYERt1_aop2_genotypes                0.20      0.10     0.01     0.40 1.00
logLYERt1_AOP2_genotypes               -0.08      0.09    -0.26     0.10 1.00
logLYERt1_temp                         -0.12      0.04    -0.21    -0.04 1.00
logLYERt1_logBiomass_g_t1              -0.07      0.21    -0.47     0.33 1.00
logPtoidt1_Intercept                   -1.94      0.53    -2.98    -0.90 1.00
logPtoidt1_aop2_genotypes               0.22      0.11     0.01     0.43 1.00
logPtoidt1_AOP2_genotypes              -0.11      0.10    -0.30     0.08 1.00
logPtoidt1_temp                        -0.15      0.06    -0.27    -0.03 1.00
logPtoidt1_logBiomass_g_t1             -1.12      0.22    -1.54    -0.68 1.00
logBRBRt1_melogBRBR_tse_logBRBRt        0.60      0.09     0.43     0.77 1.00
logBRBRt1_melogLYER_tse_logLYERt        0.13      0.12    -0.09     0.36 1.00
logBRBRt1_melogPtoid_tse_logPtoidt     -0.75      0.06    -0.87    -0.63 1.00
logLYERt1_melogLYER_tse_logLYERt        0.52      0.08     0.37     0.68 1.00
logLYERt1_melogPtoid_tse_logPtoidt     -0.71      0.05    -0.81    -0.62 1.00
logPtoidt1_melogPtoid_tse_logPtoidt     0.99      0.05     0.89     1.09 1.00
logPtoidt1_melogBRBR_tse_logBRBRt       0.08      0.08    -0.07     0.22 1.00
logPtoidt1_melogLYER_tse_logLYERt       0.45      0.10     0.26     0.63 1.00
                                    Bulk_ESS Tail_ESS
logBRBRt1_Intercept                     4900     6370
logBRBRt1_aop2_genotypes                6588     7044
logBRBRt1_AOP2_genotypes                7220     7139
logBRBRt1_temp                          4242     4897
logBRBRt1_logBiomass_g_t1               5462     7121
logLYERt1_Intercept                     3950     5754
logLYERt1_aop2_genotypes                7248     7478
logLYERt1_AOP2_genotypes                7639     7223
logLYERt1_temp                         10200     7636
logLYERt1_logBiomass_g_t1               5321     5978
logPtoidt1_Intercept                    4843     6482
logPtoidt1_aop2_genotypes               8136     7125
logPtoidt1_AOP2_genotypes               9386     7823
logPtoidt1_temp                         6076     6765
logPtoidt1_logBiomass_g_t1              6295     6744
logBRBRt1_melogBRBR_tse_logBRBRt        3617     4579
logBRBRt1_melogLYER_tse_logLYERt        3274     5075
logBRBRt1_melogPtoid_tse_logPtoidt      6193     7109
logLYERt1_melogLYER_tse_logLYERt        4190     5881
logLYERt1_melogPtoid_tse_logPtoidt      6360     6526
logPtoidt1_melogPtoid_tse_logPtoidt     7586     6935
logPtoidt1_melogBRBR_tse_logBRBRt       5280     6362
logPtoidt1_melogLYER_tse_logLYERt       4626     6567

Family Specific Parameters: 
                 Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sigma_logBRBRt1      1.24      0.06     1.13     1.36 1.00     8049     7011
sigma_logLYERt1      0.98      0.05     0.90     1.07 1.00     9924     7548
sigma_logPtoidt1     1.02      0.05     0.94     1.12 1.00    11585     7535

Residual Correlations: 
                             Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS
rescor(logBRBRt1,logLYERt1)      0.49      0.05     0.38     0.58 1.00     7244
rescor(logBRBRt1,logPtoidt1)    -0.22      0.06    -0.34    -0.10 1.00     9870
rescor(logLYERt1,logPtoidt1)    -0.06      0.07    -0.19     0.07 1.00    10875
                             Tail_ESS
rescor(logBRBRt1,logLYERt1)      7667
rescor(logBRBRt1,logPtoidt1)     7561
rescor(logLYERt1,logPtoidt1)     7909

Samples were drawn using sampling(NUTS). For each parameter, Bulk_ESS
and Tail_ESS are effective sample size measures, and Rhat is the potential
scale reduction factor on split chains (at convergence, Rhat = 1).</code></pre>
<div id="carrying-capacity-check-1" class="section level4">
<h4>Carrying capacity check</h4>
<pre class="r"><code># BRBR carrying capacity
fixef(all.mar1.brm.unadj.noBRBRonLYER)[&quot;logBRBRt1_Intercept&quot;,&quot;Estimate&quot;] / (1 - fixef(all.mar1.brm.unadj.noBRBRonLYER)[&quot;logBRBRt1_melogBRBR_tse_logBRBRt&quot;,&quot;Estimate&quot;]) # a bit low I think</code></pre>
<pre><code>[1] 5.876553</code></pre>
<pre class="r"><code>max(log(full_df$BRBR_t1))</code></pre>
<pre><code>[1] 6.727432</code></pre>
<pre class="r"><code># LYER carrying capacity
fixef(all.mar1.brm.unadj.noBRBRonLYER)[&quot;logLYERt1_Intercept&quot;,&quot;Estimate&quot;] / (1 - fixef(all.mar1.brm.unadj.noBRBRonLYER)[&quot;logLYERt1_melogLYER_tse_logLYERt&quot;,&quot;Estimate&quot;])</code></pre>
<pre><code>[1] 7.285828</code></pre>
<pre class="r"><code>max(log(full_df$LYER_t1))</code></pre>
<pre><code>[1] 6.975414</code></pre>
</div>
<div id="structural-stability-check" class="section level4">
<h4>Structural stability check</h4>
<pre class="r"><code>stability_all.mar1.brm.unadj.noBRBRonLYER &lt;- aop2_vs_AOP2_posterior_samples_unadj(all.mar1.brm.unadj.noBRBRonLYER, n.geno = 2, temp.value = 0, logbiomass.value = 0)
stability_all.mar1.brm.unadj.noBRBRonLYER$aop2_SS_LP_BayesP # same clear inference</code></pre>
<pre><code>[1] 0.9974</code></pre>
</div>
</div>
<div id="aop2-alters-interaction-matrix" class="section level3">
<h3>AOP2 alters interaction matrix</h3>
<pre class="r"><code>## Update formula ----

# BRBR
all.BRBR.bf.xAOP2 &lt;- update(all.BRBR.bf, .~. + (me(logBRBR_t, se_logBRBRt) + me(logLYER_t, se_logLYERt) + me(logPtoid_t, se_logPtoidt)):(aop2_genotypes + AOP2_genotypes))

# LYER
all.LYER.bf.xAOP2 &lt;- update(all.LYER.bf, .~. + (me(logBRBR_t, se_logBRBRt) + me(logLYER_t, se_logLYERt) + me(logPtoid_t, se_logPtoidt)):(aop2_genotypes + AOP2_genotypes))

# Ptoid
all.Ptoid.bf.xAOP2 &lt;- update(all.Ptoid.bf, .~. + (me(logBRBR_t, se_logBRBRt) + me(logLYER_t, se_logLYERt) + me(logPtoid_t, se_logPtoidt)):(aop2_genotypes + AOP2_genotypes))

## Update priors ----
all.mv.xAOP2.priors &lt;- c(
  # same priors as before
  all.mv.priors,
  # aop2 effects on interaction matrix
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;melogBRBR_tse_logBRBRt:aop2_genotypes&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;melogLYER_tse_logLYERt:aop2_genotypes&quot;, resp = &quot;logLYERt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;melogPtoid_tse_logPtoidt:aop2_genotypes&quot;, resp = &quot;logPtoidt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;melogLYER_tse_logLYERt:aop2_genotypes&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;melogBRBR_tse_logBRBRt:aop2_genotypes&quot;, resp = &quot;logLYERt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;melogPtoid_tse_logPtoidt:aop2_genotypes&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;melogPtoid_tse_logPtoidt:aop2_genotypes&quot;, resp = &quot;logLYERt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;melogBRBR_tse_logBRBRt:aop2_genotypes&quot;, resp = &quot;logPtoidt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;melogLYER_tse_logLYERt:aop2_genotypes&quot;, resp = &quot;logPtoidt1&quot;),
  # AOP2 effects on interaction matrix
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;melogBRBR_tse_logBRBRt:AOP2_genotypes&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;melogLYER_tse_logLYERt:AOP2_genotypes&quot;, resp = &quot;logLYERt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;melogPtoid_tse_logPtoidt:AOP2_genotypes&quot;, resp = &quot;logPtoidt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;melogLYER_tse_logLYERt:AOP2_genotypes&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;melogBRBR_tse_logBRBRt:AOP2_genotypes&quot;, resp = &quot;logLYERt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;melogPtoid_tse_logPtoidt:AOP2_genotypes&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;melogPtoid_tse_logPtoidt:AOP2_genotypes&quot;, resp = &quot;logLYERt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;melogBRBR_tse_logBRBRt:AOP2_genotypes&quot;, resp = &quot;logPtoidt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;melogLYER_tse_logLYERt:AOP2_genotypes&quot;, resp = &quot;logPtoidt1&quot;))

## Fit model ----
all.mar1.brm.unadj.xAOP2 &lt;- brm(
  data = full_df,
  formula = mvbf(all.BRBR.bf.xAOP2, all.LYER.bf.xAOP2, all.Ptoid.bf.xAOP2) + set_rescor(TRUE),
  iter = 6000, # increased to avoid Bulk ESS warnings
  save_pars = save_pars(latent = TRUE, all = TRUE),
  prior = all.mv.xAOP2.priors,
  file = &quot;output/all.mar1.brm.unadj.xAOP2.rds&quot;)

all.mar1.brm.unadj.xAOP2</code></pre>
<pre><code> Family: MV(gaussian, gaussian, gaussian) 
  Links: mu = identity; sigma = identity
         mu = identity; sigma = identity
         mu = identity; sigma = identity 
Formula: log(BRBR_t1) ~ Intercept + me(logBRBR_t, se_logBRBRt) + me(logLYER_t, se_logLYERt) + me(logPtoid_t, se_logPtoidt) + aop2_genotypes + AOP2_genotypes + temp + log(Biomass_g_t1) + (1 | p | Cage) + me(logBRBR_t, se_logBRBRt):aop2_genotypes + me(logBRBR_t, se_logBRBRt):AOP2_genotypes + me(logLYER_t, se_logLYERt):aop2_genotypes + me(logLYER_t, se_logLYERt):AOP2_genotypes + me(logPtoid_t, se_logPtoidt):aop2_genotypes + me(logPtoid_t, se_logPtoidt):AOP2_genotypes - 1 
         log(LYER_t1) ~ Intercept + me(logLYER_t, se_logLYERt) + me(logBRBR_t, se_logBRBRt) + me(logPtoid_t, se_logPtoidt) + aop2_genotypes + AOP2_genotypes + temp + log(Biomass_g_t1) + (1 | p | Cage) + me(logBRBR_t, se_logBRBRt):aop2_genotypes + me(logBRBR_t, se_logBRBRt):AOP2_genotypes + me(logLYER_t, se_logLYERt):aop2_genotypes + me(logLYER_t, se_logLYERt):AOP2_genotypes + me(logPtoid_t, se_logPtoidt):aop2_genotypes + me(logPtoid_t, se_logPtoidt):AOP2_genotypes - 1 
         log(Ptoid_t1) ~ Intercept + me(logPtoid_t, se_logPtoidt) + me(logBRBR_t, se_logBRBRt) + me(logLYER_t, se_logLYERt) + aop2_genotypes + AOP2_genotypes + temp + log(Biomass_g_t1) + (1 | p | Cage) + me(logBRBR_t, se_logBRBRt):aop2_genotypes + me(logBRBR_t, se_logBRBRt):AOP2_genotypes + me(logLYER_t, se_logLYERt):aop2_genotypes + me(logLYER_t, se_logLYERt):AOP2_genotypes + me(logPtoid_t, se_logPtoidt):aop2_genotypes + me(logPtoid_t, se_logPtoidt):AOP2_genotypes - 1 
   Data: full_df (Number of observations: 264) 
Samples: 4 chains, each with iter = 6000; warmup = 3000; thin = 1;
         total post-warmup samples = 12000

Group-Level Effects: 
~Cage (Number of levels: 60) 
                                              Estimate Est.Error l-95% CI
sd(logBRBRt1_Intercept)                           0.19      0.13     0.01
sd(logLYERt1_Intercept)                           0.16      0.10     0.01
sd(logPtoidt1_Intercept)                          0.07      0.06     0.00
cor(logBRBRt1_Intercept,logLYERt1_Intercept)     -0.09      0.49    -0.90
cor(logBRBRt1_Intercept,logPtoidt1_Intercept)     0.10      0.51    -0.86
cor(logLYERt1_Intercept,logPtoidt1_Intercept)    -0.03      0.50    -0.89
                                              u-95% CI Rhat Bulk_ESS Tail_ESS
sd(logBRBRt1_Intercept)                           0.46 1.00     2421     5049
sd(logLYERt1_Intercept)                           0.38 1.00     2268     4082
sd(logPtoidt1_Intercept)                          0.21 1.00     7963     5988
cor(logBRBRt1_Intercept,logLYERt1_Intercept)      0.84 1.00     3997     6271
cor(logBRBRt1_Intercept,logPtoidt1_Intercept)     0.91 1.00    11321     8313
cor(logLYERt1_Intercept,logPtoidt1_Intercept)     0.86 1.00    11121     9465

Population-Level Effects: 
                                                   Estimate Est.Error l-95% CI
logBRBRt1_Intercept                                    2.35      0.67     1.02
logBRBRt1_aop2_genotypes                              -0.00      0.39    -0.78
logBRBRt1_AOP2_genotypes                              -0.04      0.38    -0.79
logBRBRt1_temp                                        -0.44      0.08    -0.60
logBRBRt1_logBiomass_g_t1                             -0.03      0.25    -0.53
logLYERt1_Intercept                                    2.85      0.62     1.64
logLYERt1_aop2_genotypes                               0.09      0.35    -0.60
logLYERt1_AOP2_genotypes                               0.53      0.35    -0.15
logLYERt1_temp                                         0.01      0.06    -0.11
logLYERt1_logBiomass_g_t1                             -0.05      0.21    -0.45
logPtoidt1_Intercept                                  -1.72      0.65    -3.00
logPtoidt1_aop2_genotypes                              0.00      0.38    -0.73
logPtoidt1_AOP2_genotypes                             -0.02      0.37    -0.76
logPtoidt1_temp                                       -0.15      0.06    -0.27
logPtoidt1_logBiomass_g_t1                            -1.15      0.22    -1.58
logBRBRt1_melogBRBR_tse_logBRBRt                       0.81      0.16     0.48
logBRBRt1_melogLYER_tse_logLYERt                      -0.04      0.17    -0.37
logBRBRt1_melogPtoid_tse_logPtoidt                    -0.81      0.12    -1.03
logBRBRt1_melogBRBR_tse_logBRBRt:aop2_genotypes       -0.01      0.10    -0.20
logBRBRt1_melogBRBR_tse_logBRBRt:AOP2_genotypes       -0.07      0.09    -0.25
logBRBRt1_melogLYER_tse_logLYERt:aop2_genotypes       -0.03      0.10    -0.22
logBRBRt1_melogLYER_tse_logLYERt:AOP2_genotypes        0.08      0.10    -0.11
logBRBRt1_melogPtoid_tse_logPtoidt:aop2_genotypes      0.10      0.07    -0.04
logBRBRt1_melogPtoid_tse_logPtoidt:AOP2_genotypes     -0.03      0.07    -0.17
logLYERt1_melogLYER_tse_logLYERt                       0.12      0.15    -0.17
logLYERt1_melogBRBR_tse_logBRBRt                       0.55      0.13     0.29
logLYERt1_melogPtoid_tse_logPtoidt                    -0.67      0.09    -0.85
logLYERt1_melogBRBR_tse_logBRBRt:aop2_genotypes       -0.13      0.08    -0.28
logLYERt1_melogBRBR_tse_logBRBRt:AOP2_genotypes       -0.15      0.07    -0.29
logLYERt1_melogLYER_tse_logLYERt:aop2_genotypes        0.10      0.08    -0.06
logLYERt1_melogLYER_tse_logLYERt:AOP2_genotypes        0.06      0.08    -0.10
logLYERt1_melogPtoid_tse_logPtoidt:aop2_genotypes      0.09      0.06    -0.03
logLYERt1_melogPtoid_tse_logPtoidt:AOP2_genotypes     -0.11      0.06    -0.22
logPtoidt1_melogPtoid_tse_logPtoidt                    0.98      0.10     0.78
logPtoidt1_melogBRBR_tse_logBRBRt                      0.33      0.14     0.06
logPtoidt1_melogLYER_tse_logLYERt                      0.19      0.15    -0.11
logPtoidt1_melogBRBR_tse_logBRBRt:aop2_genotypes      -0.13      0.08    -0.28
logPtoidt1_melogBRBR_tse_logBRBRt:AOP2_genotypes      -0.11      0.08    -0.26
logPtoidt1_melogLYER_tse_logLYERt:aop2_genotypes       0.13      0.09    -0.03
logPtoidt1_melogLYER_tse_logLYERt:AOP2_genotypes       0.10      0.08    -0.06
logPtoidt1_melogPtoid_tse_logPtoidt:aop2_genotypes     0.05      0.06    -0.07
logPtoidt1_melogPtoid_tse_logPtoidt:AOP2_genotypes    -0.05      0.06    -0.16
                                                   u-95% CI Rhat Bulk_ESS
logBRBRt1_Intercept                                    3.65 1.00     8249
logBRBRt1_aop2_genotypes                               0.76 1.00     9743
logBRBRt1_AOP2_genotypes                               0.70 1.00     9109
logBRBRt1_temp                                        -0.29 1.00     6432
logBRBRt1_logBiomass_g_t1                              0.47 1.00    11375
logLYERt1_Intercept                                    4.07 1.00     7045
logLYERt1_aop2_genotypes                               0.79 1.00     9122
logLYERt1_AOP2_genotypes                               1.22 1.00     7954
logLYERt1_temp                                         0.13 1.00     6593
logLYERt1_logBiomass_g_t1                              0.36 1.00    10690
logPtoidt1_Intercept                                  -0.44 1.00     8446
logPtoidt1_aop2_genotypes                              0.74 1.00     9598
logPtoidt1_AOP2_genotypes                              0.71 1.00     8895
logPtoidt1_temp                                       -0.03 1.00    10674
logPtoidt1_logBiomass_g_t1                            -0.71 1.00    12565
logBRBRt1_melogBRBR_tse_logBRBRt                       1.12 1.00     4384
logBRBRt1_melogLYER_tse_logLYERt                       0.30 1.00     4727
logBRBRt1_melogPtoid_tse_logPtoidt                    -0.58 1.00     6268
logBRBRt1_melogBRBR_tse_logBRBRt:aop2_genotypes        0.18 1.00     5734
logBRBRt1_melogBRBR_tse_logBRBRt:AOP2_genotypes        0.12 1.00     6401
logBRBRt1_melogLYER_tse_logLYERt:aop2_genotypes        0.17 1.00     5784
logBRBRt1_melogLYER_tse_logLYERt:AOP2_genotypes        0.27 1.00     6072
logBRBRt1_melogPtoid_tse_logPtoidt:aop2_genotypes      0.24 1.00     7313
logBRBRt1_melogPtoid_tse_logPtoidt:AOP2_genotypes      0.10 1.00     8785
logLYERt1_melogLYER_tse_logLYERt                       0.42 1.00     4436
logLYERt1_melogBRBR_tse_logBRBRt                       0.80 1.00     4427
logLYERt1_melogPtoid_tse_logPtoidt                    -0.49 1.00     6650
logLYERt1_melogBRBR_tse_logBRBRt:aop2_genotypes        0.02 1.00     5822
logLYERt1_melogBRBR_tse_logBRBRt:AOP2_genotypes       -0.00 1.00     6166
logLYERt1_melogLYER_tse_logLYERt:aop2_genotypes        0.26 1.00     6202
logLYERt1_melogLYER_tse_logLYERt:AOP2_genotypes        0.21 1.00     5591
logLYERt1_melogPtoid_tse_logPtoidt:aop2_genotypes      0.20 1.00     7299
logLYERt1_melogPtoid_tse_logPtoidt:AOP2_genotypes     -0.00 1.00     9162
logPtoidt1_melogPtoid_tse_logPtoidt                    1.17 1.00     7012
logPtoidt1_melogBRBR_tse_logBRBRt                      0.60 1.00     4759
logPtoidt1_melogLYER_tse_logLYERt                      0.48 1.00     4981
logPtoidt1_melogBRBR_tse_logBRBRt:aop2_genotypes       0.03 1.00     6650
logPtoidt1_melogBRBR_tse_logBRBRt:AOP2_genotypes       0.03 1.00     6903
logPtoidt1_melogLYER_tse_logLYERt:aop2_genotypes       0.29 1.00     6382
logPtoidt1_melogLYER_tse_logLYERt:AOP2_genotypes       0.27 1.00     6471
logPtoidt1_melogPtoid_tse_logPtoidt:aop2_genotypes     0.18 1.00     8181
logPtoidt1_melogPtoid_tse_logPtoidt:AOP2_genotypes     0.07 1.00     9300
                                                   Tail_ESS
logBRBRt1_Intercept                                    8641
logBRBRt1_aop2_genotypes                               8766
logBRBRt1_AOP2_genotypes                               8472
logBRBRt1_temp                                         7380
logBRBRt1_logBiomass_g_t1                              9052
logLYERt1_Intercept                                    7962
logLYERt1_aop2_genotypes                               9173
logLYERt1_AOP2_genotypes                               8537
logLYERt1_temp                                         7007
logLYERt1_logBiomass_g_t1                              9333
logPtoidt1_Intercept                                   8150
logPtoidt1_aop2_genotypes                              8694
logPtoidt1_AOP2_genotypes                              8905
logPtoidt1_temp                                        9495
logPtoidt1_logBiomass_g_t1                             9380
logBRBRt1_melogBRBR_tse_logBRBRt                       6291
logBRBRt1_melogLYER_tse_logLYERt                       6595
logBRBRt1_melogPtoid_tse_logPtoidt                     7867
logBRBRt1_melogBRBR_tse_logBRBRt:aop2_genotypes        7981
logBRBRt1_melogBRBR_tse_logBRBRt:AOP2_genotypes        7482
logBRBRt1_melogLYER_tse_logLYERt:aop2_genotypes        7720
logBRBRt1_melogLYER_tse_logLYERt:AOP2_genotypes        7428
logBRBRt1_melogPtoid_tse_logPtoidt:aop2_genotypes      8702
logBRBRt1_melogPtoid_tse_logPtoidt:AOP2_genotypes      9321
logLYERt1_melogLYER_tse_logLYERt                       6560
logLYERt1_melogBRBR_tse_logBRBRt                       6765
logLYERt1_melogPtoid_tse_logPtoidt                     8133
logLYERt1_melogBRBR_tse_logBRBRt:aop2_genotypes        8428
logLYERt1_melogBRBR_tse_logBRBRt:AOP2_genotypes        7545
logLYERt1_melogLYER_tse_logLYERt:aop2_genotypes        7801
logLYERt1_melogLYER_tse_logLYERt:AOP2_genotypes        7431
logLYERt1_melogPtoid_tse_logPtoidt:aop2_genotypes      8099
logLYERt1_melogPtoid_tse_logPtoidt:AOP2_genotypes      8485
logPtoidt1_melogPtoid_tse_logPtoidt                    7899
logPtoidt1_melogBRBR_tse_logBRBRt                      7047
logPtoidt1_melogLYER_tse_logLYERt                      7167
logPtoidt1_melogBRBR_tse_logBRBRt:aop2_genotypes       8171
logPtoidt1_melogBRBR_tse_logBRBRt:AOP2_genotypes       8276
logPtoidt1_melogLYER_tse_logLYERt:aop2_genotypes       7798
logPtoidt1_melogLYER_tse_logLYERt:AOP2_genotypes       8083
logPtoidt1_melogPtoid_tse_logPtoidt:aop2_genotypes     8776
logPtoidt1_melogPtoid_tse_logPtoidt:AOP2_genotypes     8767

Family Specific Parameters: 
                 Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sigma_logBRBRt1      1.23      0.06     1.12     1.35 1.00    11864     9993
sigma_logLYERt1      0.93      0.05     0.85     1.02 1.00     8898     8868
sigma_logPtoidt1     1.02      0.05     0.93     1.11 1.00    14445     9705

Residual Correlations: 
                             Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS
rescor(logBRBRt1,logLYERt1)      0.48      0.05     0.37     0.57 1.00     9700
rescor(logBRBRt1,logPtoidt1)    -0.24      0.06    -0.36    -0.11 1.00    12108
rescor(logLYERt1,logPtoidt1)    -0.10      0.06    -0.22     0.03 1.00    14456
                             Tail_ESS
rescor(logBRBRt1,logLYERt1)      9690
rescor(logBRBRt1,logPtoidt1)     9228
rescor(logLYERt1,logPtoidt1)     9204

Samples were drawn using sampling(NUTS). For each parameter, Bulk_ESS
and Tail_ESS are effective sample size measures, and Rhat is the potential
scale reduction factor on split chains (at convergence, Rhat = 1).</code></pre>
<div id="carrying-capacity-check-2" class="section level4">
<h4>Carrying capacity check</h4>
<p><em>B. brassicae</em>:</p>
<pre class="r"><code># baseline
r_BRBR.xAOP2.all &lt;- fixef(all.mar1.brm.unadj.xAOP2)[&quot;logBRBRt1_Intercept&quot;,&quot;Estimate&quot;]
intra_BRBR.xAOP2.all &lt;- fixef(all.mar1.brm.unadj.xAOP2)[&quot;logBRBRt1_melogBRBR_tse_logBRBRt&quot;,&quot;Estimate&quot;]
K_BRBR.xAOP2.all_base &lt;- r_BRBR.xAOP2.all / (1 - intra_BRBR.xAOP2.all) # too high

# aop2 effect
aop2_r_BRBR.xAOP2.all &lt;- fixef(all.mar1.brm.unadj.xAOP2)[&quot;logBRBRt1_aop2_genotypes&quot;,&quot;Estimate&quot;]
aop2_intra_BRBR.xAOP2.all &lt;- fixef(all.mar1.brm.unadj.xAOP2)[&quot;logBRBRt1_melogBRBR_tse_logBRBRt:aop2_genotypes&quot;,&quot;Estimate&quot;]
K_BRBR.xAOP2.all_aop2 &lt;- (r_BRBR.xAOP2.all + aop2_r_BRBR.xAOP2.all) / (1 - (intra_BRBR.xAOP2.all + aop2_intra_BRBR.xAOP2.all)) 

# AOP2 effect
AOP2_r_BRBR.xAOP2.all &lt;- fixef(all.mar1.brm.unadj.xAOP2)[&quot;logBRBRt1_AOP2_genotypes&quot;,&quot;Estimate&quot;]
AOP2_intra_BRBR.xAOP2.all &lt;- fixef(all.mar1.brm.unadj.xAOP2)[&quot;logBRBRt1_melogBRBR_tse_logBRBRt:AOP2_genotypes&quot;,&quot;Estimate&quot;]
K_BRBR.xAOP2.all_AOP2 &lt;- (r_BRBR.xAOP2.all + AOP2_r_BRBR.xAOP2.all) / (1 - (intra_BRBR.xAOP2.all + AOP2_intra_BRBR.xAOP2.all)) 

# compare aop2 vs baseline carrying capacity
K_BRBR.xAOP2.all_base - K_BRBR.xAOP2.all_aop2 &gt; 0 # base &gt; aop2? Says it is, which doesn&#39;t make sense. </code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code># compare AOP2 vs baseline 
K_BRBR.xAOP2.all_base - K_BRBR.xAOP2.all_AOP2 &gt; 0 # base &gt; AOP2? Says it is, which is expected</code></pre>
<pre><code>[1] TRUE</code></pre>
<p><em>L. erysimi</em>:</p>
<pre class="r"><code># baseline
r_LYER.xAOP2.all &lt;- fixef(all.mar1.brm.unadj.xAOP2)[&quot;logLYERt1_Intercept&quot;,&quot;Estimate&quot;]
intra_LYER.xAOP2.all &lt;- fixef(all.mar1.brm.unadj.xAOP2)[&quot;logLYERt1_melogLYER_tse_logLYERt&quot;,&quot;Estimate&quot;]
K_LYER.xAOP2.all_base &lt;- r_LYER.xAOP2.all / (1 - intra_LYER.xAOP2.all) # too low

# aop2 effect
aop2_r_LYER.xAOP2.all &lt;- fixef(all.mar1.brm.unadj.xAOP2)[&quot;logLYERt1_aop2_genotypes&quot;,&quot;Estimate&quot;]
aop2_intra_LYER.xAOP2.all &lt;- fixef(all.mar1.brm.unadj.xAOP2)[&quot;logLYERt1_melogLYER_tse_logLYERt:aop2_genotypes&quot;,&quot;Estimate&quot;]
K_LYER.xAOP2.all_aop2 &lt;- (r_LYER.xAOP2.all + aop2_r_LYER.xAOP2.all) / (1 - (intra_LYER.xAOP2.all + aop2_intra_LYER.xAOP2.all))

# AOP2 effect
AOP2_r_LYER.xAOP2.all &lt;- fixef(all.mar1.brm.unadj.xAOP2)[&quot;logLYERt1_AOP2_genotypes&quot;,&quot;Estimate&quot;]
AOP2_intra_LYER.xAOP2.all &lt;- fixef(all.mar1.brm.unadj.xAOP2)[&quot;logLYERt1_melogLYER_tse_logLYERt:AOP2_genotypes&quot;,&quot;Estimate&quot;]
K_LYER.xAOP2.all_AOP2 &lt;- (r_LYER.xAOP2.all + AOP2_r_LYER.xAOP2.all) / (1 - (intra_LYER.xAOP2.all + AOP2_intra_LYER.xAOP2.all))

# compare aop2 vs baseline carrying capacity
K_LYER.xAOP2.all_base - K_LYER.xAOP2.all_aop2 &gt; 0 # base &gt; aop2? Says it isn&#39;t, which makes sense. </code></pre>
<pre><code>[1] FALSE</code></pre>
<pre class="r"><code># compare AOP2 vs baseline 
K_LYER.xAOP2.all_base - K_LYER.xAOP2.all_AOP2 &gt; 0 # base &gt; AOP2? Says it isn&#39;t, but I would expect it to be</code></pre>
<pre><code>[1] FALSE</code></pre>
<p>This model gives unrealistic estimates for <em>B. brassicae</em>’s carrying capacity (ridiculously high, 176410 individuals) and <em>L. erysimi</em>’s carrying capacity (way too low, 25 individuals). Moreover, this model predicts that adding genotypes with a null <em>AOP2</em><span class="math inline">\(-\)</span> allele to the plant population will actually <strong>decrease</strong> the aphids carrying capacity, despite the documented positive effect on plant growth. All of these biological predictions are unreasonable, so we don’t consider this model further.</p>
</div>
</div>
<div id="mar2-model" class="section level3">
<h3>MAR(2) model</h3>
<pre class="r"><code>## update formula
all.BRBR.bf.ar2 &lt;- update(all.BRBR.bf, .~. + me(logBRBR_t0, se_logBRBRt) + me(logLYER_t0, se_logLYERt) + me(logPtoid_t0, se_logPtoidt))
all.LYER.bf.ar2 &lt;- update(all.LYER.bf, .~. + me(logBRBR_t0, se_logBRBRt) + me(logLYER_t0, se_logLYERt) + me(logPtoid_t0, se_logPtoidt))
all.Ptoid.bf.ar2 &lt;- update(all.Ptoid.bf, .~. + me(logBRBR_t0, se_logBRBRt) + me(logLYER_t0, se_logLYERt) + me(logPtoid_t0, se_logPtoidt))

## fit model
all.mar1.brm.unadj.ar2 &lt;- brm(
  data = full_df,
  formula = mvbf(all.BRBR.bf.ar2, all.LYER.bf.ar2, all.Ptoid.bf.ar2) + set_rescor(TRUE),
  iter = 5000,
  save_pars = save_pars(latent = TRUE, all = TRUE),
  prior = all.mv.priors,
  file = &quot;output/all.mar1.brm.unadj.ar2.lag.rds&quot;)</code></pre>
<div id="carrying-capacity-check-3" class="section level4">
<h4>Carrying capacity check</h4>
<pre class="r"><code># BRBR carrying capacity
fixef(all.mar1.brm.unadj.ar2)[&quot;logBRBRt1_Intercept&quot;,&quot;Estimate&quot;] / (1 - fixef(all.mar1.brm.unadj.ar2)[&quot;logBRBRt1_melogBRBR_tse_logBRBRt&quot;,&quot;Estimate&quot;]) # way too high</code></pre>
<pre><code>[1] 12.56733</code></pre>
<pre class="r"><code># LYER carrying capacity
fixef(all.mar1.brm.unadj)[&quot;logLYERt1_Intercept&quot;,&quot;Estimate&quot;] / (1 - fixef(all.mar1.brm.unadj)[&quot;logLYERt1_melogLYER_tse_logLYERt&quot;,&quot;Estimate&quot;]) # way too low</code></pre>
<pre><code>[1] 4.675535</code></pre>
</div>
<div id="structural-stability-check-1" class="section level4">
<h4>Structural stability check</h4>
<pre class="r"><code>stability_all.mar1.brm.unadj.ar2 &lt;- aop2_vs_AOP2_posterior_samples_unadj(all.mar1.brm.unadj.ar2, n.geno = 2, temp.value = 0, logbiomass.value = 0)
stability_all.mar1.brm.unadj.ar2$aop2_SS_LP_BayesP # same inference!</code></pre>
<pre><code>[1] 0.9942</code></pre>
</div>
<div id="reproduce-bayesian-r2-in-table-s4-bayesian-r2" class="section level4">
<h4>Reproduce Bayesian R2 in Table S4 Bayesian R2</h4>
<pre class="r"><code>## MAR(1) model
bayes_R2(all.mar1.brm.unadj, newdata = BRBR_predict_df, resp = &quot;logBRBRt1&quot;)</code></pre>
<pre><code>             Estimate  Est.Error      Q2.5     Q97.5
R2logBRBRt1 0.6458233 0.02396662 0.5958197 0.6883213</code></pre>
<pre class="r"><code>bayes_R2(all.mar1.brm.unadj, newdata = LYER_predict_df, resp = &quot;logLYERt1&quot;)</code></pre>
<pre><code>             Estimate Est.Error     Q2.5    Q97.5
R2logLYERt1 0.4868739  0.032469 0.424568 0.550468</code></pre>
<pre class="r"><code>bayes_R2(all.mar1.brm.unadj, newdata = Ptoid_predict_df, resp = &quot;logPtoidt1&quot;)</code></pre>
<pre><code>              Estimate  Est.Error      Q2.5     Q97.5
R2logPtoidt1 0.6288823 0.01184835 0.6004538 0.6462984</code></pre>
<pre class="r"><code>## MAR(2) model 
bayes_R2(all.mar1.brm.unadj.ar2, newdata = BRBR_predict_df, resp = &quot;logBRBRt1&quot;)</code></pre>
<pre><code>             Estimate  Est.Error      Q2.5     Q97.5
R2logBRBRt1 0.7280533 0.01367774 0.6985861 0.7522238</code></pre>
<pre class="r"><code>bayes_R2(all.mar1.brm.unadj.ar2, newdata = LYER_predict_df, resp = &quot;logLYERt1&quot;)</code></pre>
<pre><code>             Estimate  Est.Error      Q2.5     Q97.5
R2logLYERt1 0.5115962 0.03208708 0.4474643 0.5734133</code></pre>
<pre class="r"><code>bayes_R2(all.mar1.brm.unadj.ar2, newdata = Ptoid_predict_df, resp = &quot;logPtoidt1&quot;)</code></pre>
<pre><code>              Estimate  Est.Error      Q2.5     Q97.5
R2logPtoidt1 0.5971516 0.01486279 0.5649386 0.6233025</code></pre>
</div>
</div>
<div id="adjusted-data" class="section level3">
<h3>Adjusted data</h3>
<pre class="r"><code>## update formula ----

# BRBR
all.BRBR.bf.adj &lt;- bf(log(BRBR_t1) ~ 0 + Intercept + (me(logBRBR_t, se_logBRBRt) + me(logLYER_t_adj, se_logLYERt) + me(logPtoid_t_adj, se_logPtoidt)) + aop2_genotypes + AOP2_genotypes + temp + log(Biomass_g_t1) + (1|p|Cage))

# LYER
all.LYER.bf.adj &lt;- bf(log(LYER_t1_adj) ~ 0 + Intercept + (me(logLYER_t_adj, se_logLYERt) + me(logBRBR_t, se_logBRBRt) + me(logPtoid_t_adj, se_logPtoidt)) + aop2_genotypes + AOP2_genotypes + temp + log(Biomass_g_t1) + (1|p|Cage))

# Ptoid
all.Ptoid.bf.adj &lt;- bf(log(Ptoid_t1_adj) ~ 0 + Intercept + me(logPtoid_t_adj, se_logPtoidt) + me(logBRBR_t, se_logBRBRt) + me(logLYER_t_adj, se_logLYERt) + aop2_genotypes + temp + AOP2_genotypes + log(Biomass_g_t1) + (1|p|Cage))

## update priors ----
all.mv.priors.adj &lt;- c(
  # aop2 and AOP2 effects
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;aop2_genotypes&quot;, resp = &quot;logBRBRt1&quot;), 
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;AOP2_genotypes&quot;, resp = &quot;logBRBRt1&quot;), 
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;aop2_genotypes&quot;, resp = &quot;logLYERt1adj&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;AOP2_genotypes&quot;, resp = &quot;logLYERt1adj&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;aop2_genotypes&quot;, resp = &quot;logPtoidt1adj&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;AOP2_genotypes&quot;, resp = &quot;logPtoidt1adj&quot;),
  # biomass effects
  set_prior(prior.AphidBiomass, class = &quot;b&quot;, coef = &quot;logBiomass_g_t1&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.AphidBiomass, class = &quot;b&quot;, coef = &quot;logBiomass_g_t1&quot;, resp = &quot;logLYERt1adj&quot;),
  set_prior(prior.PtoidBiomass, class = &quot;b&quot;, coef = &quot;logBiomass_g_t1&quot;, resp = &quot;logPtoidt1adj&quot;),
  # temp effects
  set_prior(prior.temp, class = &quot;b&quot;, coef = &quot;temp&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.temp, class = &quot;b&quot;, coef = &quot;temp&quot;, resp = &quot;logLYERt1adj&quot;),
  set_prior(prior.temp, class = &quot;b&quot;, coef = &quot;temp&quot;, resp = &quot;logPtoidt1adj&quot;),
  # baseline growth rates
  set_prior(prior.r.BRBR, class = &quot;b&quot;, coef = &quot;Intercept&quot;, resp = &quot;logBRBRt1&quot;), 
  set_prior(prior.r.LYER, class = &quot;b&quot;, coef = &quot;Intercept&quot;, resp = &quot;logLYERt1adj&quot;),
  set_prior(prior.r.Ptoid, class = &quot;b&quot;, coef = &quot;Intercept&quot;, resp = &quot;logPtoidt1adj&quot;),
  # intraspecific effects
  set_prior(prior.intra.BRBR, class = &quot;b&quot;, coef = &quot;melogBRBR_tse_logBRBRt&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.intra.LYER, class = &quot;b&quot;, coef = &quot;melogLYER_t_adjse_logLYERt&quot;, resp = &quot;logLYERt1adj&quot;), 
  set_prior(prior.intra.Ptoid, class = &quot;b&quot;, coef = &quot;melogPtoid_t_adjse_logPtoidt&quot;, resp = &quot;logPtoidt1adj&quot;),
  # negative interspecific effects
  set_prior(prior.LYERonBRBR, class = &quot;b&quot;, coef = &quot;melogLYER_t_adjse_logLYERt&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.BRBRonLYER, class = &quot;b&quot;, coef = &quot;melogBRBR_tse_logBRBRt&quot;, resp = &quot;logLYERt1adj&quot;),
  set_prior(prior.PtoidonBRBR, class = &quot;b&quot;, coef = &quot;melogPtoid_t_adjse_logPtoidt&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.PtoidonLYER, class = &quot;b&quot;, coef = &quot;melogPtoid_t_adjse_logPtoidt&quot;, resp = &quot;logLYERt1adj&quot;),
  # positive interspecific effects
  set_prior(prior.BRBRonPtoid, class = &quot;b&quot;, coef = &quot;melogBRBR_tse_logBRBRt&quot;, resp = &quot;logPtoidt1adj&quot;),
  set_prior(prior.LYERonPtoid, class = &quot;b&quot;, coef = &quot;melogLYER_t_adjse_logLYERt&quot;, resp = &quot;logPtoidt1adj&quot;),
  # random effects
  set_prior(prior.random.effects, class = &quot;sd&quot;, resp = &quot;logBRBRt1&quot;), 
  set_prior(prior.random.effects, class = &quot;sd&quot;, resp = &quot;logLYERt1adj&quot;), 
  set_prior(prior.random.effects, class = &quot;sd&quot;, resp = &quot;logPtoidt1adj&quot;))

## fit model
all.mar1.brm.adj &lt;- brm(
  data = full_df,
  formula = mvbf(all.BRBR.bf.adj, all.LYER.bf.adj, all.Ptoid.bf.adj) + set_rescor(TRUE),
  iter = 5000,
  save_pars = save_pars(latent = TRUE, all = TRUE),
  prior = all.mv.priors.adj,
  file = &quot;output/all.mar1.brm.adj.rds&quot;)</code></pre>
<div id="structural-stability-check-2" class="section level4">
<h4>Structural stability check</h4>
<pre class="r"><code>stability_all.mar1.brm.adj &lt;- aop2_vs_AOP2_posterior_samples_adj(all.mar1.brm.adj, n.geno = 2, temp.value = 0, logbiomass.value = 0) 
stability_all.mar1.brm.adj$aop2_SS_LP_BayesP # same inference</code></pre>
<pre><code>[1] 0.979</code></pre>
</div>
</div>
<div id="aphid-intrinsic-growth-rates" class="section level3">
<h3>Aphid intrinsic growth rates</h3>
<p>Confirmation with independent data (not used in model for all species) that <em>AOP</em><span class="math inline">\(-\)</span> increases the intrinsic growth rate of the aphids.</p>
<div id="unadjusted-data" class="section level4">
<h4>Unadjusted data</h4>
<pre class="r"><code># BRBR
initial.BRBR.bf &lt;- bf(log(BRBR_t1) ~ 0 + Intercept + offset(log(4)) + aop2_genotypes + AOP2_genotypes + temp) 

# LYER
initial.LYER.bf &lt;- bf(log(LYER_t1) ~ 0 + Intercept + offset(log(4)) + aop2_genotypes + AOP2_genotypes + temp)

# Prior
initial.mv.prior &lt;- c(
  # aop2 effects
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;aop2_genotypes&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;AOP2_genotypes&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;aop2_genotypes&quot;, resp = &quot;logLYERt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;AOP2_genotypes&quot;, resp = &quot;logLYERt1&quot;),
  # temp effects
  set_prior(prior.temp, class = &quot;b&quot;, coef = &quot;temp&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.temp, class = &quot;b&quot;, coef = &quot;temp&quot;, resp = &quot;logLYERt1&quot;),
  # growth rates
  set_prior(prior.r.BRBR, class = &quot;b&quot;, coef = &quot;Intercept&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.r.LYER, class = &quot;b&quot;, coef = &quot;Intercept&quot;, resp = &quot;logLYERt1&quot;))

# fit model
initial.mar1.brm.unadj &lt;- brm(
  data = filter(aphids_only_df, Week == 0),
  formula = mvbf(initial.BRBR.bf, initial.LYER.bf) + set_rescor(TRUE),
  iter = 5000, 
  save_pars = save_pars(latent = TRUE, all = TRUE),
  prior = initial.mv.prior,
  file = &quot;output/initial.mar1.brm.unadj.rds&quot;) 

# get posteriors
ps_initial.mar1.brm.unadj &lt;- posterior_samples(initial.mar1.brm.unadj, pars = c(&quot;b_logBRBRt1_aop2_genotypes&quot;,&quot;b_logBRBRt1_AOP2_genotypes&quot;,&quot;b_logLYERt1_aop2_genotypes&quot;,&quot;b_logLYERt1_AOP2_genotypes&quot;)) %&gt;%
  mutate(aop2_vs_AOP2_r_BRBR = b_logBRBRt1_aop2_genotypes - b_logBRBRt1_AOP2_genotypes,
         aop2_vs_AOP2_r_LYER = b_logLYERt1_aop2_genotypes - b_logLYERt1_AOP2_genotypes)

# r BRBR, aop2 vs AOP2
median(ps_initial.mar1.brm.unadj$aop2_vs_AOP2_r_BRBR)</code></pre>
<pre><code>[1] 0.3298602</code></pre>
<pre class="r"><code>quantile(ps_initial.mar1.brm.unadj$aop2_vs_AOP2_r_BRBR, probs = c(0.025,0.975))</code></pre>
<pre><code>      2.5%      97.5% 
0.06062928 0.58563572 </code></pre>
<pre class="r"><code># r LYER, aop2 vs AOP2
median(ps_initial.mar1.brm.unadj$aop2_vs_AOP2_r_LYER)</code></pre>
<pre><code>[1] 1.014212</code></pre>
<pre class="r"><code>quantile(ps_initial.mar1.brm.unadj$aop2_vs_AOP2_r_LYER, probs = c(0.025,0.975))</code></pre>
<pre><code>     2.5%     97.5% 
0.5872244 1.4224264 </code></pre>
</div>
<div id="adjusted-data-1" class="section level4">
<h4>Adjusted data</h4>
<pre class="r"><code># LYER
initial.LYER.bf.adj &lt;- bf(log(LYER_t1_adj) ~ 0 + Intercept + offset(log(4)) + aop2_genotypes + AOP2_genotypes + temp)

# Priors
initial.mv.prior.adj &lt;- c(
  # aop2 effects
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;aop2_genotypes&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;AOP2_genotypes&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;aop2_genotypes&quot;, resp = &quot;logLYERt1adj&quot;),
  set_prior(prior.AOP2, class = &quot;b&quot;, coef = &quot;AOP2_genotypes&quot;, resp = &quot;logLYERt1adj&quot;),
  # temp effects
  set_prior(prior.temp, class = &quot;b&quot;, coef = &quot;temp&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.temp, class = &quot;b&quot;, coef = &quot;temp&quot;, resp = &quot;logLYERt1adj&quot;),
  # growth rates
  set_prior(prior.r.BRBR, class = &quot;b&quot;, coef = &quot;Intercept&quot;, resp = &quot;logBRBRt1&quot;),
  set_prior(prior.r.LYER, class = &quot;b&quot;, coef = &quot;Intercept&quot;, resp = &quot;logLYERt1adj&quot;))

# fit model
initial.mar1.brm.adj &lt;- brm(
  data = filter(aphids_only_df, Week == 0),
  formula = mvbf(initial.BRBR.bf, initial.LYER.bf.adj) + set_rescor(TRUE),
  iter = 5000, 
  save_pars = save_pars(latent = TRUE, all = TRUE),
  prior = initial.mv.prior.adj,
  file = &quot;output/initial.mar1.brm.adj.rds&quot;) 

ps_initial.mar1.brm.adj &lt;- posterior_samples(initial.mar1.brm.adj, pars = c(&quot;b_logBRBRt1_aop2_genotypes&quot;,&quot;b_logBRBRt1_AOP2_genotypes&quot;,&quot;b_logLYERt1adj_aop2_genotypes&quot;,&quot;b_logLYERt1adj_AOP2_genotypes&quot;)) %&gt;%
  mutate(aop2_vs_AOP2_r_BRBR = b_logBRBRt1_aop2_genotypes - b_logBRBRt1_AOP2_genotypes,
         aop2_vs_AOP2_r_LYER = b_logLYERt1adj_aop2_genotypes - b_logLYERt1adj_AOP2_genotypes)

# r BRBR, aop2 vs AOP2
median(ps_initial.mar1.brm.adj$aop2_vs_AOP2_r_BRBR)</code></pre>
<pre><code>[1] 0.3369524</code></pre>
<pre class="r"><code>quantile(ps_initial.mar1.brm.adj$aop2_vs_AOP2_r_BRBR, probs = c(0.025,0.975))</code></pre>
<pre><code>     2.5%     97.5% 
0.0749945 0.5938041 </code></pre>
<pre class="r"><code># r LYER, aop2 vs AOP2
median(ps_initial.mar1.brm.adj$aop2_vs_AOP2_r_LYER)</code></pre>
<pre><code>[1] 0.4574658</code></pre>
<pre class="r"><code>quantile(ps_initial.mar1.brm.adj$aop2_vs_AOP2_r_LYER, probs = c(0.025,0.975))</code></pre>
<pre><code>     2.5%     97.5% 
0.2832546 0.6327860 </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.7 LTS

Matrix products: default
BLAS:   /usr/lib/libblas/libblas.so.3.6.0
LAPACK: /usr/lib/lapack/liblapack.so.3.6.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] bayesplot_1.8.0 rgl_0.106.8     knitr_1.33      tidybayes_2.3.1
 [5] matlib_0.9.4    cowplot_1.1.1   forcats_0.5.1   stringr_1.4.0  
 [9] dplyr_1.0.6     purrr_0.3.4     readr_1.4.0     tidyr_1.1.3    
[13] tibble_3.1.2    ggplot2_3.3.3   tidyverse_1.3.1 brms_2.15.0    
[17] Rcpp_1.0.6      RCurl_1.98-1.3  MASS_7.3-54     workflowr_1.6.2

loaded via a namespace (and not attached):
  [1] readxl_1.3.1            backports_1.2.1         plyr_1.8.6             
  [4] igraph_1.2.6            svUnit_1.0.6            splines_4.1.0          
  [7] crosstalk_1.1.1         rstantools_2.1.1        inline_0.3.19          
 [10] digest_0.6.27           htmltools_0.5.1.1       viridis_0.6.1          
 [13] rsconnect_0.8.18        fansi_0.5.0             magrittr_2.0.1         
 [16] openxlsx_4.2.4          modelr_0.1.8            RcppParallel_5.1.4     
 [19] matrixStats_0.59.0      xts_0.12.1              prettyunits_1.1.1      
 [22] colorspace_2.0-1        rvest_1.0.0             ggdist_2.4.1           
 [25] haven_2.4.1             xfun_0.23               callr_3.7.0            
 [28] crayon_1.4.1            jsonlite_1.7.2          lme4_1.1-27            
 [31] zoo_1.8-9               glue_1.4.2              gtable_0.3.0           
 [34] V8_3.4.2                distributional_0.2.2    car_3.0-10             
 [37] pkgbuild_1.2.0          rstan_2.21.2            abind_1.4-5            
 [40] scales_1.1.1            mvtnorm_1.1-2           DBI_1.1.1              
 [43] miniUI_0.1.1.1          viridisLite_0.4.0       xtable_1.8-3           
 [46] foreign_0.8-81          stats4_4.1.0            StanHeaders_2.21.0-7   
 [49] DT_0.18                 htmlwidgets_1.5.3       httr_1.4.2             
 [52] threejs_0.3.3           arrayhelpers_1.1-0      ellipsis_0.3.2         
 [55] farver_2.1.0            pkgconfig_2.0.3         loo_2.4.1              
 [58] sass_0.4.0              dbplyr_2.1.1            utf8_1.2.1             
 [61] labeling_0.4.2          tidyselect_1.1.1        rlang_0.4.11           
 [64] manipulateWidget_0.11.0 reshape2_1.4.4          later_1.2.0            
 [67] munsell_0.5.0           cellranger_1.1.0        tools_4.1.0            
 [70] cli_2.5.0               generics_0.1.0          broom_0.7.6            
 [73] ggridges_0.5.3          evaluate_0.14           fastmap_1.1.0          
 [76] yaml_2.2.1              processx_3.5.2          fs_1.5.0               
 [79] zip_2.2.0               nlme_3.1-152            whisker_0.4            
 [82] mime_0.10               projpred_2.0.2          xml2_1.3.2             
 [85] compiler_4.1.0          shinythemes_1.2.0       rstudioapi_0.13        
 [88] curl_4.3.1              gamm4_0.2-6             reprex_2.0.0           
 [91] bslib_0.2.5.1           stringi_1.6.2           highr_0.9              
 [94] ps_1.6.0                Brobdingnag_1.2-6       lattice_0.20-44        
 [97] Matrix_1.3-4            nloptr_1.2.2.2          markdown_1.1           
[100] shinyjs_2.0.0           vctrs_0.3.8             pillar_1.6.1           
[103] lifecycle_1.0.0         jquerylib_0.1.4         bridgesampling_1.1-2   
[106] data.table_1.14.0       bitops_1.0-7            httpuv_1.6.1           
[109] R6_2.5.0                promises_1.2.0.1        gridExtra_2.3          
[112] rio_0.5.26              codetools_0.2-18        boot_1.3-28            
[115] colourpicker_1.1.0      gtools_3.9.2            assertthat_0.2.1       
[118] rprojroot_2.0.2         withr_2.4.2             shinystan_2.5.0        
[121] mgcv_1.8-36             parallel_4.1.0          hms_1.1.0              
[124] grid_4.1.0              coda_0.19-4             minqa_1.2.4            
[127] rmarkdown_2.8           carData_3.0-4           git2r_0.28.0           
[130] shiny_1.6.0             lubridate_1.7.10        base64enc_0.1-3        
[133] dygraphs_1.1.1.6       </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
