---
title: "Community persistence"
author: "Matthew Barbour"
date: "`r Sys.Date()`"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, message = FALSE)

# load required libraries
library(tidyverse) # for data management
library(cowplot)
library(clubSandwich)
library(kableExtra)

# set ggplot theme
theme_set(theme_cowplot())
```

# Setup

```{r}
# Load and manage data
df <- read_csv("data/arabidopsis_clean_df.csv") %>%
  # renaming for brevity
  rename(cage = Cage,
         com = Composition,
         week = Week,
         temp = Temperature,
         rich = Richness) %>%
  mutate(cage = as.character(cage),
         fweek = factor(ifelse(week < 10, paste("0", week, sep=""), week)),
         temp = ifelse(temp=="20 C", 0, 3)) %>% # set to 3 so that 1 C = 1 genotype
  arrange(cage, week)

# focus on last week of experiment
df17 <- df %>%
  # counter information is not relevant (because it is the same), so we summarise across it
  group_by(cage, fweek, week, temp, rich, Col, gsm1, AOP2, AOP2.gsoh, com) %>%
  summarise_at(vars(BRBR_Survival, LYER_Survival, Mummy_Ptoids_Survival), list(mean)) %>%
  ungroup() %>%
  # we need the dataset to go through week 17, rather than removing cages as they transition
  # to a collapsed community as in `state_df`
  mutate(BRBR = ifelse(is.na(BRBR_Survival) == T, 0, BRBR_Survival),
         LYER = ifelse(is.na(LYER_Survival) == T, 0, LYER_Survival),
         Ptoid = ifelse(is.na(Mummy_Ptoids_Survival) == T, 0, Mummy_Ptoids_Survival)) %>%
  filter(week == 17) %>%
  mutate(# calculate persistence
    Persistence = BRBR + LYER + Ptoid,
    rich - 1, # baseline of 1 genotype
    # define orthogonal constrasts to test for above-average allele effects.
    # aop2_vs_AOP2 must be included first before testing for mam1_vs_MAM1 and gsoh_vs_GSOH
    aop2_vs_AOP2 = Col + gsm1 - AOP2 - AOP2.gsoh,
    mam1_vs_MAM1 = gsm1 - Col,
    gsoh_vs_GSOH = AOP2.gsoh - AOP2) 
# note that lowercase denotes the null (non-functional) version of the allele
# wherease capital indicates the functional form
```

```{r}
# source in useful functions for analyses
source('code/glm-ftest.R') # ANOVA GLM
```

# Analysis

Below, we reproduce the analysis of deviance for food-web persistence presented in Table S1 in the Supplementary Material.

The analysis below tests for a general effect of genetic diversity (`rich`) as well as above-average effects of allelic differences at *AOP2*, *MAM1*, and *GSOH*. It also tests for an effect of temperature (`temp`) and whether temperature modifies any of these genetic effects.

```{r}
glm.ftest.v2(
  model = glm(data = df17,
              family = quasibinomial(link = "cloglog"),
              formula = terms(Persistence/3 ~
                                temp + rich + aop2_vs_AOP2 + mam1_vs_MAM1 + gsoh_vs_GSOH + com +
                                temp:(rich + aop2_vs_AOP2 + mam1_vs_MAM1 + gsoh_vs_GSOH) + temp:com,
                              keep.order = T)),
  test.formula = list(
    c("temp","temp:com"),
    c("rich","com"),
    c("aop2_vs_AOP2","com"),
    c("mam1_vs_MAM1","com"),
    c("gsoh_vs_GSOH","com"),
    c("temp:rich","temp:com"),
    c("temp:aop2_vs_AOP2","temp:com"),
    c("temp:mam1_vs_MAM1","temp:com"),
    c("temp:gsoh_vs_GSOH","temp:com"))
)[[3]] %>%
  select(Source = treatment,
         `df (Source)` = num_df,
         `df (Error)` = den_df,
         Deviance = deviance,
         `Mean Deviance` = mean_deviance,
         F = F, P = P, Error = error)
```

We observe a clear effect of genetic diversity and an above-average contribution of *AOP2* to food-web persistence. Temperature did not have a clear effect on food-web persistence and didn't modify genetic effects.

# Effect sizes

Genetic diversity increased the probability of a species persisting by `r exp(coef(glm(data = df17, family = quasibinomial(link = "cloglog"), formula = Persistence/3 ~ temp + rich))["rich"]) - 1`%.

```{r}
# calculate change in probability
exp(coef(glm(data = df17, family = quasibinomial(link = "cloglog"), formula = Persistence/3 ~ temp + rich))["rich"]) - 1
```

# Keystone gene plot (Fig. 1 in main text)

```{r}
gene_CI <- conf_int(
  glm(data = df17,
      family = quasibinomial(link = "cloglog"),
      formula = terms(Persistence/3 ~
                        temp + rich + aop2_vs_AOP2 + mam1_vs_MAM1 + gsoh_vs_GSOH,
                      keep.order = T)),
  vcov = "CR2",
  test = "naive-t",
  cluster = df17$com,
  coefs = c("aop2_vs_AOP2","mam1_vs_MAM1","gsoh_vs_GSOH")
) %>%
  data.frame() %>%
  rownames_to_column(var = "term") %>%
  mutate(gene = factor(c("AOP2","MAM1","GSOH"), levels = c("AOP2","MAM1","GSOH")))#ordered = T))

# replacing the average genotype with a genotype that has an aop2 (vs AOP2) allele
# results in a 28% increase in probability of a species persisting
exp(gene_CI$beta[1])-1

# this plot makes clear that AOP2 gene has an above-average effect on community persistence.
ggplot(gene_CI, aes(x = gene, y = exp(beta)-1)) +
  geom_point(size = 5) +
  geom_linerange(aes(ymax = exp(beta + SE)-1, ymin = exp(beta - SE)-1), size = 1.5) +
  geom_linerange(aes(ymax = exp(CI_U)-1, ymin = exp(CI_L)-1)) +
  geom_hline(yintercept = 0, linetype = "dotted") +
  scale_x_discrete(labels = c(expression(italic(AOP2)),expression(italic(MAM1)),expression(italic(GSOH)))) +
  #scale_y_continuous("Probability of species persisting ()") +
  scale_y_continuous(name = expression("Effect on food-web persistence "(Delta~"%"))) + # "odds"
  xlab("")
# ggsave(filename = "figures/keystone-gene.pdf", height = 6, width = 8)
```

Let's visualize the effect of particular alleles within *AOP2*.

```{r}
aop2_CI <- conf_int(
  glm(data = df17,
      family = quasibinomial(link = "cloglog"),
      formula = terms(Persistence/3 ~
                        temp + I(AOP2 + AOP2.gsoh) + I(Col + gsm1),
                      keep.order = T)),
  vcov = "CR2",
  test = "naive-t",
  cluster = df17$com,
  coefs = c("I(AOP2 + AOP2.gsoh)","I(Col + gsm1)")
) %>%
  data.frame() %>%
  rownames_to_column(var = "term") %>%
  mutate(allele = c("AOP2","aop2"))
exp(aop2_CI$beta[2])-1 # 80% increase

# get the effect of each genotype
mean_geno <- conf_int(
  glm(data = df17,
      family = quasibinomial(link = "cloglog"),
      formula = terms(Persistence/3 ~
                        temp + AOP2 + AOP2.gsoh + Col + gsm1,
                      keep.order = T)),
  vcov = "CR2",
  test = "naive-t",
  cluster = df17$com,
  coefs = c("AOP2","AOP2.gsoh","Col","gsm1")
) %>%
  data.frame() %>%
  rownames_to_column(var = "term") %>%
  mutate(allele = c("AOP2","AOP2","aop2","aop2"),
         term = factor(term, levels = c("Col","gsm1","AOP2","AOP2.gsoh"), labels = c("Col","gsm1","AOP2","AOP2/gsoh")))

# adding a genotype with an aop2 allele to the population doubles the likelihood of species persistence
ggplot(aop2_CI, aes(x = allele, y = exp(beta)-1)) +
  geom_point(size = 5) +
  geom_point(data = mean_geno, aes(color = term), size = 5, position = position_dodge(width = 0.3)) +
  geom_linerange(aes(ymax = exp(beta + SE)-1, ymin = exp(beta - SE)-1), size = 1.5) +
  geom_linerange(aes(ymax = exp(CI_U)-1, ymin = exp(CI_L)-1)) +
  geom_hline(yintercept = 0, linetype = "dotted") +
  scale_x_discrete(labels = c("AOP2\u2013","AOP2+")) +
  scale_y_continuous(expression("Effect on food-web persistence "(Delta~"%"))) +
  xlab("Allele") + #xlab(expression("Allele at "~italic(AOP2)~"gene")) +
  scale_color_manual(values = c("darkgreen","steelblue","darkorange","firebrick1"), name = "")#, name = "Genotype")
```

# Does *AOP2*$-$ explain genetic diversity effect?

```{r}
glm.ftest.v2(
  model = glm(data = df17,
              family = quasibinomial(link = "cloglog"),
              formula = terms(Persistence/3 ~
                                temp + I(Col + gsm1) + rich + com +
                                temp + temp:com,
                              keep.order = T)),
  test.formula = list(
    c("temp","temp:com"),
    c("I(Col + gsm1)","com"),
    #c("I(AOP2 + AOP2.gsoh)","com"), # rich has a very strong effect if you only include AOP2 + AOP2.gsoh
    c("rich","com"))
)[[3]] %>%
  select(Source = treatment,
         `df (Source)` = num_df,
         `df (Error)` = den_df,
         Deviance = deviance,
         `Mean Deviance` = mean_deviance,
         F = F, P = P, Error = error)
```

The above model suggests that the effect of genetic diversity is explained entirely by the increased probability of having genotypes with an *AOP2*$-$ in the plant population.

Note that if instead we modeled the effect of *AOP2*+ before genetic diversity, we still observe a clear effect of genetic diversity.

```{r}
glm.ftest.v2(
  model = glm(data = df17,
              family = quasibinomial(link = "cloglog"),
              formula = terms(Persistence/3 ~
                                temp + I(AOP2 + AOP2.gsoh) + rich + com +
                                temp + temp:com,
                              keep.order = T)),
  test.formula = list(
    c("temp","temp:com"),
    #c("I(Col + gsm1)","com"),
    c("I(AOP2 + AOP2.gsoh)","com"),
    c("rich","com"))
)[[3]] %>%
  select(Source = treatment,
         `df (Source)` = num_df,
         `df (Error)` = den_df,
         Deviance = deviance,
         `Mean Deviance` = mean_deviance,
         F = F, P = P, Error = error)
```

# Save analysis

Write out an `.RData` file to use for creating the Supplementary Material Results.

```{r}
# save.image(file = "output/community-persistence-keystone.RData")
```
